<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Biopsy Quality Control | Longitudinal measures of RNA-seq expression and disease activity in FSHD muscle biopsies</title>
  <meta name="description" content="Bioinformatics and statistics analysis for the logitudinal study of RNA-seq expression and disear acitivity in FSHD muscle bipsies" />
  <meta name="generator" content="bookdown 0.14 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Biopsy Quality Control | Longitudinal measures of RNA-seq expression and disease activity in FSHD muscle biopsies" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Bioinformatics and statistics analysis for the logitudinal study of RNA-seq expression and disear acitivity in FSHD muscle bipsies" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Biopsy Quality Control | Longitudinal measures of RNA-seq expression and disease activity in FSHD muscle biopsies" />
  
  <meta name="twitter:description" content="Bioinformatics and statistics analysis for the logitudinal study of RNA-seq expression and disear acitivity in FSHD muscle bipsies" />
  

<meta name="author" content="Chao-Jen Wong" />


<meta name="date" content="2019-12-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="references.html"/>

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">RNA-seq data analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#package-structure"><i class="fa fa-check"></i><b>1.1</b> Package structure</a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#datasets"><i class="fa fa-check"></i><b>1.1.1</b> Datasets</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#geo"><i class="fa fa-check"></i><b>1.2</b> GEO</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#system-requirment"><i class="fa fa-check"></i><b>1.3</b> System requirment</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="prepro.html"><a href="prepro.html"><i class="fa fa-check"></i><b>2</b> Preprocessing and gene counts</a><ul>
<li class="chapter" data-level="2.1" data-path="prepro.html"><a href="prepro.html#shell-script-for-pre-processing"><i class="fa fa-check"></i><b>2.1</b> Shell script for pre-processing</a></li>
<li class="chapter" data-level="2.2" data-path="prepro.html"><a href="prepro.html#annotation-txdb-package"><i class="fa fa-check"></i><b>2.2</b> Annotation TxDb package</a></li>
<li class="chapter" data-level="2.3" data-path="prepro.html"><a href="prepro.html#r-script-for-gene-counts"><i class="fa fa-check"></i><b>2.3</b> R Script for gene counts</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="part1.html"><a href="part1.html"><i class="fa fa-check"></i><b>3</b> Follow-up visit</a><ul>
<li class="chapter" data-level="3.1" data-path="part1.html"><a href="part1.html#the-follow-up-visit-rna-seq-dataset"><i class="fa fa-check"></i><b>3.1</b> The follow-up visit RNA-seq dataset</a></li>
<li class="chapter" data-level="3.2" data-path="part1.html"><a href="part1.html#dux4-and-other-marker-scores"><i class="fa fa-check"></i><b>3.2</b> DUX4 and other marker scores</a><ul>
<li class="chapter" data-level="3.2.1" data-path="part1.html"><a href="part1.html#group-samples-by-dux4-score"><i class="fa fa-check"></i><b>3.2.1</b> Group samples by DUX4 score</a></li>
<li class="chapter" data-level="3.2.2" data-path="part1.html"><a href="part1.html#heatmap-of-the-discriminative-genes"><i class="fa fa-check"></i><b>3.2.2</b> Heatmap of the discriminative genes</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="part1.html"><a href="part1.html#dux4-candidate-biomarkers-are-eleveated-in-fshd"><i class="fa fa-check"></i><b>3.3</b> DUX4 candidate biomarkers are eleveated in FSHD</a></li>
<li class="chapter" data-level="3.4" data-path="part1.html"><a href="part1.html#correlation-of-dux4-score-to-mri-and-histopathology"><i class="fa fa-check"></i><b>3.4</b> Correlation of DUX4 score to MRI and histopathology</a><ul>
<li class="chapter" data-level="3.4.1" data-path="part1.html"><a href="part1.html#dux4-and-pathology-score"><i class="fa fa-check"></i><b>3.4.1</b> DUX4 and pathology score</a></li>
<li class="chapter" data-level="3.4.2" data-path="part1.html"><a href="part1.html#dux4-and-stir-signal"><i class="fa fa-check"></i><b>3.4.2</b> DUX4 and STIR signal</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="part1.html"><a href="part1.html#dux4-score-and-t1-fat-fraction"><i class="fa fa-check"></i><b>3.5</b> DUX4 score and T1 fat fraction</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="part2.html"><a href="part2.html"><i class="fa fa-check"></i><b>4</b> Changes over two visits</a><ul>
<li class="chapter" data-level="4.1" data-path="part2.html"><a href="part2.html#pathology-scores"><i class="fa fa-check"></i><b>4.1</b> Pathology scores</a></li>
<li class="chapter" data-level="4.2" data-path="part2.html"><a href="part2.html#dux4-and-other-marker-scores-1"><i class="fa fa-check"></i><b>4.2</b> DUX4 and other marker scores</a><ul>
<li class="chapter" data-level="4.2.1" data-path="part2.html"><a href="part2.html#dux4"><i class="fa fa-check"></i><b>4.2.1</b> DUX4</a></li>
<li class="chapter" data-level="4.2.2" data-path="part2.html"><a href="part2.html#other-markers"><i class="fa fa-check"></i><b>4.2.2</b> Other markers</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="part2.html"><a href="part2.html#pathology-and-dux4-scores-changes"><i class="fa fa-check"></i><b>4.3</b> Pathology and DUX4 scores changes</a></li>
<li class="chapter" data-level="4.4" data-path="part2.html"><a href="part2.html#markers-by-visits-and-stir-ratings"><i class="fa fa-check"></i><b>4.4</b> Markers by visits and STIR ratings</a><ul>
<li class="chapter" data-level="4.4.1" data-path="part2.html"><a href="part2.html#dux4-1"><i class="fa fa-check"></i><b>4.4.1</b> DUX4</a></li>
<li class="chapter" data-level="4.4.2" data-path="part2.html"><a href="part2.html#other-markers-1"><i class="fa fa-check"></i><b>4.4.2</b> Other markers</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="applications.html"><a href="applications.html"><i class="fa fa-check"></i><b>5</b> Applications</a><ul>
<li class="chapter" data-level="5.1" data-path="applications.html"><a href="applications.html#example-one"><i class="fa fa-check"></i><b>5.1</b> Example one</a></li>
<li class="chapter" data-level="5.2" data-path="applications.html"><a href="applications.html#example-two"><i class="fa fa-check"></i><b>5.2</b> Example two</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>6</b> Final Words</a></li>
<li class="chapter" data-level="7" data-path="pax7.html"><a href="pax7.html"><i class="fa fa-check"></i><b>7</b> PAX7 targeted genes</a><ul>
<li class="chapter" data-level="7.1" data-path="pax7.html"><a href="pax7.html#pa7-scores-and-disease-activity"><i class="fa fa-check"></i><b>7.1</b> PA7 scores and disease activity</a><ul>
<li class="chapter" data-level="7.1.1" data-path="pax7.html"><a href="pax7.html#boxplot-of-pax7-score"><i class="fa fa-check"></i><b>7.1.1</b> Boxplot of PAX7 score</a></li>
<li class="chapter" data-level="7.1.2" data-path="pax7.html"><a href="pax7.html#heapmap-of-pax7-targeted-gene-expression"><i class="fa fa-check"></i><b>7.1.2</b> Heapmap of PAX7 targeted gene expression</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="pax7.html"><a href="pax7.html#view-from-a-different-angle"><i class="fa fa-check"></i><b>7.2</b> View from a different angle</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html"><i class="fa fa-check"></i><b>8</b> Immune Cell Infiltration</a><ul>
<li class="chapter" data-level="8.1" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html#immune-cell-markers"><i class="fa fa-check"></i><b>8.1</b> Immune cell markers</a></li>
<li class="chapter" data-level="8.2" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html#immune-cell-type-scores-by-tpm-per-samples"><i class="fa fa-check"></i><b>8.2</b> Immune cell type scores by TPM per Samples</a></li>
<li class="chapter" data-level="8.3" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html#immune-cell-marker-scores-by-logfc-per-fshd-classes"><i class="fa fa-check"></i><b>8.3</b> Immune cell marker scores by logFC per FSHD classes</a></li>
<li class="chapter" data-level="8.4" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html#frequency-of-significance"><i class="fa fa-check"></i><b>8.4</b> Frequency of significance</a></li>
<li class="chapter" data-level="8.5" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html#average-rlog-per-sample"><i class="fa fa-check"></i><b>8.5</b> Average rlog per sample</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="9" data-path="biopsy-qc.html"><a href="biopsy-qc.html"><i class="fa fa-check"></i><b>9</b> Biopsy Quality Control</a><ul>
<li class="chapter" data-level="9.1" data-path="biopsy-qc.html"><a href="biopsy-qc.html#blood-fat-and-muscle-cell-markers"><i class="fa fa-check"></i><b>9.1</b> Blood, fat and muscle cell markers</a></li>
<li class="chapter" data-level="9.2" data-path="biopsy-qc.html"><a href="biopsy-qc.html#define-score"><i class="fa fa-check"></i><b>9.2</b> Define score</a></li>
<li class="chapter" data-level="9.3" data-path="biopsy-qc.html"><a href="biopsy-qc.html#visualization"><i class="fa fa-check"></i><b>9.3</b> Visualization</a></li>
<li class="chapter" data-level="9.4" data-path="biopsy-qc.html"><a href="biopsy-qc.html#determine-outliers"><i class="fa fa-check"></i><b>9.4</b> Determine “outliers”</a></li>
<li class="chapter" data-level="9.5" data-path="biopsy-qc.html"><a href="biopsy-qc.html#determin-bad-biopsies"><i class="fa fa-check"></i><b>9.5</b> Determin “bad” biopsies?</a></li>
<li class="chapter" data-level="9.6" data-path="biopsy-qc.html"><a href="biopsy-qc.html#tables"><i class="fa fa-check"></i><b>9.6</b> Tables</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Longitudinal measures of RNA-seq expression and disease activity in FSHD muscle biopsies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="biopsy_qc" class="section level1">
<h1><span class="header-section-number">Chapter 9</span> Biopsy Quality Control</h1>
<p>Blood, fat, muscle markers and scored by <span class="math inline">\(\log_{10}(TPM + 1)\)</span>.</p>
<div id="blood-fat-and-muscle-cell-markers" class="section level2">
<h2><span class="header-section-number">9.1</span> Blood, fat and muscle cell markers</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">markers &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">cell_type=</span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">&quot;blood&quot;</span>, <span class="dv">3</span>), <span class="kw">rep</span>(<span class="st">&quot;fat&quot;</span>, <span class="dv">3</span>), <span class="kw">rep</span>(<span class="st">&quot;muscle&quot;</span>, <span class="dv">3</span>)),
                  <span class="dt">gene_name=</span><span class="kw">c</span>(<span class="st">&quot;HBA1&quot;</span>, <span class="st">&quot;HBA2&quot;</span>, <span class="st">&quot;HBB&quot;</span>,
                              <span class="st">&quot;FASN&quot;</span>, <span class="st">&quot;LEP&quot;</span>, <span class="st">&quot;SCD&quot;</span>,
                              <span class="st">&quot;ACTA1&quot;</span>, <span class="st">&quot;TNNT3&quot;</span>, <span class="st">&quot;MYH1&quot;</span>))  <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">gencode_id =</span> <span class="kw">get_ensembl</span>(gene_name, <span class="dt">rse=</span>sanitized.dds)) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">cell_type =</span> <span class="kw">factor</span>(cell_type))</code></pre></div>
<pre><code>## HBA1 HBA2 HBB FASN LEP SCD ACTA1 TNNT3 MYH1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr<span class="op">::</span><span class="kw">kable</span>(markers)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">cell_type</th>
<th align="left">gene_name</th>
<th align="left">gencode_id</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">blood</td>
<td align="left">HBA1</td>
<td align="left">ENSG00000206172.8</td>
</tr>
<tr class="even">
<td align="left">blood</td>
<td align="left">HBA2</td>
<td align="left">ENSG00000188536.12</td>
</tr>
<tr class="odd">
<td align="left">blood</td>
<td align="left">HBB</td>
<td align="left">ENSG00000244734.3</td>
</tr>
<tr class="even">
<td align="left">fat</td>
<td align="left">FASN</td>
<td align="left">ENSG00000169710.7</td>
</tr>
<tr class="odd">
<td align="left">fat</td>
<td align="left">LEP</td>
<td align="left">ENSG00000174697.4</td>
</tr>
<tr class="even">
<td align="left">fat</td>
<td align="left">SCD</td>
<td align="left">ENSG00000099194.5</td>
</tr>
<tr class="odd">
<td align="left">muscle</td>
<td align="left">ACTA1</td>
<td align="left">ENSG00000143632.14</td>
</tr>
<tr class="even">
<td align="left">muscle</td>
<td align="left">TNNT3</td>
<td align="left">ENSG00000130595.16</td>
</tr>
<tr class="odd">
<td align="left">muscle</td>
<td align="left">MYH1</td>
<td align="left">ENSG00000109061.9</td>
</tr>
</tbody>
</table>
</div>
<div id="define-score" class="section level2">
<h2><span class="header-section-number">9.2</span> Define score</h2>
<p>For each cell type, the marker score for each sample is <span class="math inline">\(\frac{1}{n}\Sigma_{i=1}^{n}\log_{10}(TPM_i + 1)\)</span>, where <span class="math inline">\(i\)</span> denotes genes and <span class="math inline">\(n\)</span> is the number of genes of the cell type.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lv &lt;-<span class="st"> </span><span class="kw">levels</span>(markers<span class="op">$</span>cell_type)
celltype_tpm &lt;-<span class="st"> </span><span class="kw">sapply</span>(lv, <span class="cf">function</span>(type) {
   id &lt;-<span class="st"> </span>markers <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(cell_type <span class="op">==</span><span class="st"> </span>type) <span class="op">%&gt;%</span>
<span class="st">      </span><span class="kw">pull</span>(gencode_id)
   sub &lt;-<span class="st"> </span>sanitized.dds[id]
   tpm_score &lt;-<span class="st"> </span><span class="kw">colMeans</span>(<span class="kw">log10</span>(<span class="kw">assays</span>(sub)[[<span class="st">&quot;TPM&quot;</span>]]<span class="op">+</span><span class="dv">1</span>))
})

celltype_tpm &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(celltype_tpm) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var=</span><span class="st">&quot;sample_name&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">add_column</span>(<span class="dt">pheno_type =</span> sanitized.dds<span class="op">$</span>pheno_type,
             <span class="dt">classes =</span> sanitized.dds<span class="op">$</span>new_cluster_name,
             <span class="dt">classes_color =</span> sanitized.dds<span class="op">$</span>cluster_color) </code></pre></div>
</div>
<div id="visualization" class="section level2">
<h2><span class="header-section-number">9.3</span> Visualization</h2>
<p>The blood, fat and muscle content scores are given to each sample. Boxplot below illustrates that the FSHD biopsies categorized in the Muscle-Low class are fatter and “blooder” and have less muscle content than the rest of the biopsies.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tmp &lt;-<span class="st"> </span>celltype_tpm <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">gather</span>(<span class="dt">key=</span>cell_type, <span class="dt">value=</span>score, <span class="op">-</span>sample_name,
                               <span class="op">-</span>pheno_type, <span class="op">-</span>classes, <span class="op">-</span>classes_color) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">cell_type =</span> <span class="kw">factor</span>(cell_type))

color_manual &lt;-<span class="st"> </span>celltype_tpm <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(classes) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">summarise</span>(<span class="dt">color=</span><span class="kw">unique</span>(classes_color)) 

<span class="kw">ggplot</span>(tmp, <span class="kw">aes</span>(<span class="dt">x=</span>classes, <span class="dt">y=</span>score, <span class="dt">color=</span>classes)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_boxplot</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="st"> </span>cell_type, <span class="dt">scale=</span><span class="st">&quot;free&quot;</span>,  <span class="dt">nrow=</span><span class="dv">2</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(), <span class="dt">legend.justification=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">0</span>), <span class="dt">legend.position=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">0</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>color_manual<span class="op">$</span>color) <span class="op">+</span>
<span class="st">  </span><span class="kw">labs</span>(<span class="dt">y=</span><span class="st">&quot;Avg. log10(TPM +1)&quot;</span>, <span class="dt">x=</span><span class="st">&quot;FSHD classes&quot;</span>, 
       <span class="dt">title=</span><span class="st">&quot;Blood, fat, muscle content score by TPM&quot;</span>)</code></pre></div>
<p><img src="gitbook_files/figure-html/boxplot_for_qc-1.png" width="672" /></p>
</div>
<div id="determine-outliers" class="section level2">
<h2><span class="header-section-number">9.4</span> Determine “outliers”</h2>
<p>use permutation test? calculate the p-value by constructing mean distribution?</p>
</div>
<div id="determin-bad-biopsies" class="section level2">
<h2><span class="header-section-number">9.5</span> Determin “bad” biopsies?</h2>
</div>
<div id="tables" class="section level2">
<h2><span class="header-section-number">9.6</span> Tables</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tb &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(celltype_tpm, <span class="op">-</span>classes_color) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">arrange</span>(classes)
knitr<span class="op">::</span><span class="kw">kable</span>(tb, <span class="dt">caption=</span><span class="st">&quot;Blood, fat and muscle content index.&quot;</span>)</code></pre></div>
<table>
<caption>(#tab:kable_celltype_tpm)Blood, fat and muscle content index.</caption>
<thead>
<tr class="header">
<th align="left">sample_name</th>
<th align="right">blood</th>
<th align="right">fat</th>
<th align="right">muscle</th>
<th align="left">pheno_type</th>
<th align="left">classes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">01-0041</td>
<td align="right">2.585593</td>
<td align="right">0.1184236</td>
<td align="right">3.862732</td>
<td align="left">Control</td>
<td align="left">Control</td>
</tr>
<tr class="even">
<td align="left">01-0042</td>
<td align="right">2.604758</td>
<td align="right">0.3822024</td>
<td align="right">3.404674</td>
<td align="left">Control</td>
<td align="left">Control</td>
</tr>
<tr class="odd">
<td align="left">01-0043</td>
<td align="right">1.515110</td>
<td align="right">0.2021819</td>
<td align="right">3.643485</td>
<td align="left">Control</td>
<td align="left">Control</td>
</tr>
<tr class="even">
<td align="left">01-0044</td>
<td align="right">2.256298</td>
<td align="right">0.2361598</td>
<td align="right">3.655246</td>
<td align="left">Control</td>
<td align="left">Control</td>
</tr>
<tr class="odd">
<td align="left">01-0045</td>
<td align="right">2.485550</td>
<td align="right">0.3499366</td>
<td align="right">3.707235</td>
<td align="left">Control</td>
<td align="left">Control</td>
</tr>
<tr class="even">
<td align="left">01-0046</td>
<td align="right">3.001476</td>
<td align="right">0.2935164</td>
<td align="right">3.427649</td>
<td align="left">Control</td>
<td align="left">Control</td>
</tr>
<tr class="odd">
<td align="left">01-0048</td>
<td align="right">2.154881</td>
<td align="right">0.3454669</td>
<td align="right">3.692780</td>
<td align="left">Control</td>
<td align="left">Control</td>
</tr>
<tr class="even">
<td align="left">01-0049</td>
<td align="right">2.199308</td>
<td align="right">0.3796754</td>
<td align="right">3.562354</td>
<td align="left">Control</td>
<td align="left">Control</td>
</tr>
<tr class="odd">
<td align="left">01-0023</td>
<td align="right">1.668277</td>
<td align="right">0.5237397</td>
<td align="right">3.177025</td>
<td align="left">FSHD</td>
<td align="left">Mild</td>
</tr>
<tr class="even">
<td align="left">01-0032</td>
<td align="right">3.282519</td>
<td align="right">0.5813589</td>
<td align="right">3.790697</td>
<td align="left">FSHD</td>
<td align="left">Mild</td>
</tr>
<tr class="odd">
<td align="left">01-0033</td>
<td align="right">2.471344</td>
<td align="right">0.2054422</td>
<td align="right">3.418840</td>
<td align="left">FSHD</td>
<td align="left">Mild</td>
</tr>
<tr class="even">
<td align="left">01-0034</td>
<td align="right">2.912924</td>
<td align="right">0.2439875</td>
<td align="right">3.968929</td>
<td align="left">FSHD</td>
<td align="left">Mild</td>
</tr>
<tr class="odd">
<td align="left">01-0036</td>
<td align="right">2.292336</td>
<td align="right">0.3440663</td>
<td align="right">3.634696</td>
<td align="left">FSHD</td>
<td align="left">Mild</td>
</tr>
<tr class="even">
<td align="left">32-0011</td>
<td align="right">3.213565</td>
<td align="right">0.2619783</td>
<td align="right">3.981536</td>
<td align="left">FSHD</td>
<td align="left">Mild</td>
</tr>
<tr class="odd">
<td align="left">32-0012</td>
<td align="right">3.013254</td>
<td align="right">0.6171513</td>
<td align="right">3.480103</td>
<td align="left">FSHD</td>
<td align="left">Mild</td>
</tr>
<tr class="even">
<td align="left">32-0014</td>
<td align="right">4.299332</td>
<td align="right">0.1537942</td>
<td align="right">3.158579</td>
<td align="left">FSHD</td>
<td align="left">Mild</td>
</tr>
<tr class="odd">
<td align="left">32-0015</td>
<td align="right">2.891994</td>
<td align="right">0.3012466</td>
<td align="right">4.061947</td>
<td align="left">FSHD</td>
<td align="left">Mild</td>
</tr>
<tr class="even">
<td align="left">32-0018</td>
<td align="right">3.617742</td>
<td align="right">0.1898964</td>
<td align="right">3.597990</td>
<td align="left">FSHD</td>
<td align="left">Mild</td>
</tr>
<tr class="odd">
<td align="left">32-0019</td>
<td align="right">2.280754</td>
<td align="right">0.3573577</td>
<td align="right">3.984106</td>
<td align="left">FSHD</td>
<td align="left">Mild</td>
</tr>
<tr class="even">
<td align="left">01-0023b</td>
<td align="right">2.273485</td>
<td align="right">0.7225922</td>
<td align="right">3.462251</td>
<td align="left">FSHD</td>
<td align="left">Mild</td>
</tr>
<tr class="odd">
<td align="left">01-0026b</td>
<td align="right">3.076727</td>
<td align="right">0.3619324</td>
<td align="right">3.636448</td>
<td align="left">FSHD</td>
<td align="left">Mild</td>
</tr>
<tr class="even">
<td align="left">01-0036b</td>
<td align="right">2.757293</td>
<td align="right">0.3583505</td>
<td align="right">3.653238</td>
<td align="left">FSHD</td>
<td align="left">Mild</td>
</tr>
<tr class="odd">
<td align="left">32-0013b</td>
<td align="right">2.824210</td>
<td align="right">0.6457411</td>
<td align="right">3.511864</td>
<td align="left">FSHD</td>
<td align="left">Mild</td>
</tr>
<tr class="even">
<td align="left">32-0015b</td>
<td align="right">2.741116</td>
<td align="right">0.3704249</td>
<td align="right">4.023872</td>
<td align="left">FSHD</td>
<td align="left">Mild</td>
</tr>
<tr class="odd">
<td align="left">32-0017b</td>
<td align="right">3.553026</td>
<td align="right">0.3916796</td>
<td align="right">3.089687</td>
<td align="left">FSHD</td>
<td align="left">Mild</td>
</tr>
<tr class="even">
<td align="left">01-0021</td>
<td align="right">2.883033</td>
<td align="right">0.4052833</td>
<td align="right">3.240677</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="odd">
<td align="left">01-0022-1</td>
<td align="right">2.603183</td>
<td align="right">1.5294705</td>
<td align="right">3.530700</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="even">
<td align="left">01-0024</td>
<td align="right">3.011862</td>
<td align="right">0.4650832</td>
<td align="right">3.470005</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="odd">
<td align="left">01-0026</td>
<td align="right">2.393720</td>
<td align="right">0.4211012</td>
<td align="right">3.681180</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="even">
<td align="left">01-0030</td>
<td align="right">3.123735</td>
<td align="right">1.2808079</td>
<td align="right">3.732009</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="odd">
<td align="left">01-0035</td>
<td align="right">2.397119</td>
<td align="right">0.9189982</td>
<td align="right">3.448796</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="even">
<td align="left">01-0038</td>
<td align="right">2.478129</td>
<td align="right">0.3810781</td>
<td align="right">3.103982</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="odd">
<td align="left">32-0002</td>
<td align="right">2.912887</td>
<td align="right">0.5254690</td>
<td align="right">3.199493</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="even">
<td align="left">32-0007</td>
<td align="right">2.481612</td>
<td align="right">0.9001335</td>
<td align="right">3.186218</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="odd">
<td align="left">32-0008</td>
<td align="right">3.944364</td>
<td align="right">1.7901614</td>
<td align="right">2.653674</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="even">
<td align="left">32-0010</td>
<td align="right">3.251255</td>
<td align="right">0.2792022</td>
<td align="right">4.161168</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="odd">
<td align="left">32-0013</td>
<td align="right">3.600465</td>
<td align="right">0.9677029</td>
<td align="right">3.692289</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="even">
<td align="left">01-0022b</td>
<td align="right">3.330682</td>
<td align="right">1.7365568</td>
<td align="right">3.796104</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="odd">
<td align="left">01-0024b</td>
<td align="right">3.183899</td>
<td align="right">0.2378188</td>
<td align="right">3.327007</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="even">
<td align="left">01-0028b</td>
<td align="right">2.163574</td>
<td align="right">0.5596671</td>
<td align="right">3.756815</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="odd">
<td align="left">01-0030b</td>
<td align="right">2.371968</td>
<td align="right">0.6290769</td>
<td align="right">3.702877</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="even">
<td align="left">01-0033b</td>
<td align="right">1.301926</td>
<td align="right">0.5558120</td>
<td align="right">3.672376</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="odd">
<td align="left">01-0034b</td>
<td align="right">2.695132</td>
<td align="right">0.5194755</td>
<td align="right">4.035298</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="even">
<td align="left">01-0035b</td>
<td align="right">2.638491</td>
<td align="right">1.0832977</td>
<td align="right">3.332846</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="odd">
<td align="left">32-0006b</td>
<td align="right">3.557061</td>
<td align="right">1.0401034</td>
<td align="right">3.582340</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="even">
<td align="left">32-0009b</td>
<td align="right">2.429373</td>
<td align="right">1.5374834</td>
<td align="right">3.066060</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="odd">
<td align="left">32-0014b</td>
<td align="right">3.373019</td>
<td align="right">0.8191356</td>
<td align="right">3.226881</td>
<td align="left">FSHD</td>
<td align="left">Moderate</td>
</tr>
<tr class="even">
<td align="left">01-0027</td>
<td align="right">2.861061</td>
<td align="right">1.4552309</td>
<td align="right">3.341357</td>
<td align="left">FSHD</td>
<td align="left">IG-High</td>
</tr>
<tr class="odd">
<td align="left">32-0017</td>
<td align="right">3.645956</td>
<td align="right">1.6709002</td>
<td align="right">3.011805</td>
<td align="left">FSHD</td>
<td align="left">IG-High</td>
</tr>
<tr class="even">
<td align="left">01-0025b</td>
<td align="right">3.760612</td>
<td align="right">1.5639605</td>
<td align="right">4.035969</td>
<td align="left">FSHD</td>
<td align="left">IG-High</td>
</tr>
<tr class="odd">
<td align="left">01-0027b</td>
<td align="right">3.458966</td>
<td align="right">1.8465812</td>
<td align="right">3.437264</td>
<td align="left">FSHD</td>
<td align="left">IG-High</td>
</tr>
<tr class="even">
<td align="left">01-0029b</td>
<td align="right">4.002951</td>
<td align="right">1.8184439</td>
<td align="right">3.068390</td>
<td align="left">FSHD</td>
<td align="left">IG-High</td>
</tr>
<tr class="odd">
<td align="left">32-0007b</td>
<td align="right">2.808018</td>
<td align="right">1.7246814</td>
<td align="right">3.398351</td>
<td align="left">FSHD</td>
<td align="left">IG-High</td>
</tr>
<tr class="even">
<td align="left">32-0018b</td>
<td align="right">3.137937</td>
<td align="right">1.1417576</td>
<td align="right">2.966068</td>
<td align="left">FSHD</td>
<td align="left">IG-High</td>
</tr>
<tr class="odd">
<td align="left">01-0025</td>
<td align="right">2.279839</td>
<td align="right">0.6538198</td>
<td align="right">3.906637</td>
<td align="left">FSHD</td>
<td align="left">High</td>
</tr>
<tr class="even">
<td align="left">01-0029</td>
<td align="right">3.213923</td>
<td align="right">1.1415584</td>
<td align="right">3.201525</td>
<td align="left">FSHD</td>
<td align="left">High</td>
</tr>
<tr class="odd">
<td align="left">32-0003</td>
<td align="right">2.920859</td>
<td align="right">1.1202031</td>
<td align="right">3.604562</td>
<td align="left">FSHD</td>
<td align="left">High</td>
</tr>
<tr class="even">
<td align="left">32-0004</td>
<td align="right">2.000568</td>
<td align="right">1.2400798</td>
<td align="right">3.354000</td>
<td align="left">FSHD</td>
<td align="left">High</td>
</tr>
<tr class="odd">
<td align="left">32-0005</td>
<td align="right">4.325426</td>
<td align="right">1.2125240</td>
<td align="right">3.661033</td>
<td align="left">FSHD</td>
<td align="left">High</td>
</tr>
<tr class="even">
<td align="left">32-0006</td>
<td align="right">2.389504</td>
<td align="right">1.1906286</td>
<td align="right">3.322845</td>
<td align="left">FSHD</td>
<td align="left">High</td>
</tr>
<tr class="odd">
<td align="left">32-0009</td>
<td align="right">3.518911</td>
<td align="right">2.3004965</td>
<td align="right">2.868475</td>
<td align="left">FSHD</td>
<td align="left">High</td>
</tr>
<tr class="even">
<td align="left">32-0002b1</td>
<td align="right">3.798370</td>
<td align="right">0.6701894</td>
<td align="right">3.311355</td>
<td align="left">FSHD</td>
<td align="left">High</td>
</tr>
<tr class="odd">
<td align="left">32-0005b</td>
<td align="right">4.165111</td>
<td align="right">0.8942373</td>
<td align="right">4.057378</td>
<td align="left">FSHD</td>
<td align="left">High</td>
</tr>
<tr class="even">
<td align="left">01-0037</td>
<td align="right">3.820057</td>
<td align="right">2.5362662</td>
<td align="right">1.146418</td>
<td align="left">FSHD</td>
<td align="left">Muscle-Low</td>
</tr>
<tr class="odd">
<td align="left">32-0016</td>
<td align="right">4.296978</td>
<td align="right">2.5212618</td>
<td align="right">1.444503</td>
<td align="left">FSHD</td>
<td align="left">Muscle-Low</td>
</tr>
<tr class="even">
<td align="left">01-0037b</td>
<td align="right">3.680353</td>
<td align="right">1.3811564</td>
<td align="right">3.556671</td>
<td align="left">FSHD</td>
<td align="left">Muscle-Low</td>
</tr>
<tr class="odd">
<td align="left">32-0010b</td>
<td align="right">4.148983</td>
<td align="right">2.2940258</td>
<td align="right">1.548513</td>
<td align="left">FSHD</td>
<td align="left">Muscle-Low</td>
</tr>
<tr class="even">
<td align="left">32-0012b</td>
<td align="right">4.738981</td>
<td align="right">2.6884333</td>
<td align="right">1.356221</td>
<td align="left">FSHD</td>
<td align="left">Muscle-Low</td>
</tr>
<tr class="odd">
<td align="left">32-0016b</td>
<td align="right">4.944297</td>
<td align="right">0.8406186</td>
<td align="right">1.415840</td>
<td align="left">FSHD</td>
<td align="left">Muscle-Low</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#&#39; blood, fat and muscle mean score for each class</span>
tb2 &lt;-<span class="st"> </span>tb <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">group_by</span>(classes) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">summarise</span>(<span class="dt">blood_mean =</span> <span class="kw">mean</span>(blood),
            <span class="dt">fat_mean =</span> <span class="kw">mean</span>(fat),
            <span class="dt">muscle_mean =</span> <span class="kw">mean</span>(muscle))
knitr<span class="op">::</span><span class="kw">kable</span>(tb2)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">classes</th>
<th align="right">blood_mean</th>
<th align="right">fat_mean</th>
<th align="right">muscle_mean</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Control</td>
<td align="right">2.350372</td>
<td align="right">0.2884454</td>
<td align="right">3.619519</td>
</tr>
<tr class="even">
<td align="left">Mild</td>
<td align="right">2.892347</td>
<td align="right">0.3900435</td>
<td align="right">3.625400</td>
</tr>
<tr class="odd">
<td align="left">Moderate</td>
<td align="right">2.823931</td>
<td align="right">0.8446781</td>
<td align="right">3.481763</td>
</tr>
<tr class="even">
<td align="left">IG-High</td>
<td align="right">3.382214</td>
<td align="right">1.6030794</td>
<td align="right">3.322743</td>
</tr>
<tr class="odd">
<td align="left">High</td>
<td align="right">3.179168</td>
<td align="right">1.1581930</td>
<td align="right">3.476423</td>
</tr>
<tr class="even">
<td align="left">Muscle-Low</td>
<td align="right">4.271608</td>
<td align="right">2.0436270</td>
<td align="right">1.744694</td>
</tr>
</tbody>
</table>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="references.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["gitbook.pdf", "gitbook.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
