<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Immune Cell Infiltration | Longitudinal measures of RNA-seq expression and disease activity in FSHD muscle biopsies</title>
  <meta name="description" content="Bioinformatics and statistics analysis for the logitudinal study of RNA-seq expression and disear acitivity in FSHD muscle bipsies" />
  <meta name="generator" content="bookdown 0.14 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Immune Cell Infiltration | Longitudinal measures of RNA-seq expression and disease activity in FSHD muscle biopsies" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Bioinformatics and statistics analysis for the logitudinal study of RNA-seq expression and disear acitivity in FSHD muscle bipsies" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Immune Cell Infiltration | Longitudinal measures of RNA-seq expression and disease activity in FSHD muscle biopsies" />
  
  <meta name="twitter:description" content="Bioinformatics and statistics analysis for the logitudinal study of RNA-seq expression and disear acitivity in FSHD muscle bipsies" />
  

<meta name="author" content="Chao-Jen Wong" />


<meta name="date" content="2019-12-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="pax7.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">RNA-seq data analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#package-structure"><i class="fa fa-check"></i><b>1.1</b> Package structure</a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#datasets"><i class="fa fa-check"></i><b>1.1.1</b> Datasets</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#geo"><i class="fa fa-check"></i><b>1.2</b> GEO</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#system-requirment"><i class="fa fa-check"></i><b>1.3</b> System requirment</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="prepro.html"><a href="prepro.html"><i class="fa fa-check"></i><b>2</b> Preprocessing and gene counts</a><ul>
<li class="chapter" data-level="2.1" data-path="prepro.html"><a href="prepro.html#shell-script-for-pre-processing"><i class="fa fa-check"></i><b>2.1</b> Shell script for pre-processing</a></li>
<li class="chapter" data-level="2.2" data-path="prepro.html"><a href="prepro.html#annotation-txdb-package"><i class="fa fa-check"></i><b>2.2</b> Annotation TxDb package</a></li>
<li class="chapter" data-level="2.3" data-path="prepro.html"><a href="prepro.html#r-script-for-gene-counts"><i class="fa fa-check"></i><b>2.3</b> R Script for gene counts</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="part1.html"><a href="part1.html"><i class="fa fa-check"></i><b>3</b> Follow-up visit</a><ul>
<li class="chapter" data-level="3.1" data-path="part1.html"><a href="part1.html#the-follow-up-visit-rna-seq-dataset"><i class="fa fa-check"></i><b>3.1</b> The follow-up visit RNA-seq dataset</a></li>
<li class="chapter" data-level="3.2" data-path="part1.html"><a href="part1.html#dux4-and-other-marker-scores"><i class="fa fa-check"></i><b>3.2</b> DUX4 and other marker scores</a><ul>
<li class="chapter" data-level="3.2.1" data-path="part1.html"><a href="part1.html#group-samples-by-dux4-score"><i class="fa fa-check"></i><b>3.2.1</b> Group samples by DUX4 score</a></li>
<li class="chapter" data-level="3.2.2" data-path="part1.html"><a href="part1.html#heatmap-of-the-discriminative-genes"><i class="fa fa-check"></i><b>3.2.2</b> Heatmap of the discriminative genes</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="part1.html"><a href="part1.html#dux4-candidate-biomarkers-are-eleveated-in-fshd"><i class="fa fa-check"></i><b>3.3</b> DUX4 candidate biomarkers are eleveated in FSHD</a></li>
<li class="chapter" data-level="3.4" data-path="part1.html"><a href="part1.html#correlation-of-dux4-score-to-mri-and-histopathology"><i class="fa fa-check"></i><b>3.4</b> Correlation of DUX4 score to MRI and histopathology</a><ul>
<li class="chapter" data-level="3.4.1" data-path="part1.html"><a href="part1.html#dux4-and-pathology-score"><i class="fa fa-check"></i><b>3.4.1</b> DUX4 and pathology score</a></li>
<li class="chapter" data-level="3.4.2" data-path="part1.html"><a href="part1.html#dux4-and-stir-signal"><i class="fa fa-check"></i><b>3.4.2</b> DUX4 and STIR signal</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="part1.html"><a href="part1.html#dux4-score-and-t1-fat-fraction"><i class="fa fa-check"></i><b>3.5</b> DUX4 score and T1 fat fraction</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="part2.html"><a href="part2.html"><i class="fa fa-check"></i><b>4</b> Changes over two visits</a><ul>
<li class="chapter" data-level="4.1" data-path="part2.html"><a href="part2.html#pathology-scores"><i class="fa fa-check"></i><b>4.1</b> Pathology scores</a></li>
<li class="chapter" data-level="4.2" data-path="part2.html"><a href="part2.html#dux4-and-other-marker-scores-1"><i class="fa fa-check"></i><b>4.2</b> DUX4 and other marker scores</a><ul>
<li class="chapter" data-level="4.2.1" data-path="part2.html"><a href="part2.html#dux4"><i class="fa fa-check"></i><b>4.2.1</b> DUX4</a></li>
<li class="chapter" data-level="4.2.2" data-path="part2.html"><a href="part2.html#other-markers"><i class="fa fa-check"></i><b>4.2.2</b> Other markers</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="part2.html"><a href="part2.html#pathology-and-dux4-scores-changes"><i class="fa fa-check"></i><b>4.3</b> Pathology and DUX4 scores changes</a></li>
<li class="chapter" data-level="4.4" data-path="part2.html"><a href="part2.html#markers-by-visits-and-stir-ratings"><i class="fa fa-check"></i><b>4.4</b> Markers by visits and STIR ratings</a><ul>
<li class="chapter" data-level="4.4.1" data-path="part2.html"><a href="part2.html#dux4-1"><i class="fa fa-check"></i><b>4.4.1</b> DUX4</a></li>
<li class="chapter" data-level="4.4.2" data-path="part2.html"><a href="part2.html#other-markers-1"><i class="fa fa-check"></i><b>4.4.2</b> Other markers</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="applications.html"><a href="applications.html"><i class="fa fa-check"></i><b>5</b> Applications</a><ul>
<li class="chapter" data-level="5.1" data-path="applications.html"><a href="applications.html#example-one"><i class="fa fa-check"></i><b>5.1</b> Example one</a></li>
<li class="chapter" data-level="5.2" data-path="applications.html"><a href="applications.html#example-two"><i class="fa fa-check"></i><b>5.2</b> Example two</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>6</b> Final Words</a></li>
<li class="chapter" data-level="7" data-path="pax7.html"><a href="pax7.html"><i class="fa fa-check"></i><b>7</b> PAX7 targeted genes</a><ul>
<li class="chapter" data-level="7.1" data-path="pax7.html"><a href="pax7.html#pa7-scores-and-disease-activity"><i class="fa fa-check"></i><b>7.1</b> PA7 scores and disease activity</a><ul>
<li class="chapter" data-level="7.1.1" data-path="pax7.html"><a href="pax7.html#boxplot-of-pax7-score"><i class="fa fa-check"></i><b>7.1.1</b> Boxplot of PAX7 score</a></li>
<li class="chapter" data-level="7.1.2" data-path="pax7.html"><a href="pax7.html#heapmap-of-pax7-targeted-gene-expression"><i class="fa fa-check"></i><b>7.1.2</b> Heapmap of PAX7 targeted gene expression</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="pax7.html"><a href="pax7.html#view-from-a-different-angle"><i class="fa fa-check"></i><b>7.2</b> View from a different angle</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html"><i class="fa fa-check"></i><b>8</b> Immune Cell Infiltration</a><ul>
<li class="chapter" data-level="8.1" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html#immune-cell-markers"><i class="fa fa-check"></i><b>8.1</b> Immune cell markers</a></li>
<li class="chapter" data-level="8.2" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html#immune-cell-type-scores-by-tpm-per-samples"><i class="fa fa-check"></i><b>8.2</b> Immune cell type scores by TPM per Samples</a></li>
<li class="chapter" data-level="8.3" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html#immune-cell-marker-scores-by-logfc-per-fshd-classes"><i class="fa fa-check"></i><b>8.3</b> Immune cell marker scores by logFC per FSHD classes</a></li>
<li class="chapter" data-level="8.4" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html#frequency-of-significance"><i class="fa fa-check"></i><b>8.4</b> Frequency of significance</a></li>
<li class="chapter" data-level="8.5" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html#average-rlog-per-sample"><i class="fa fa-check"></i><b>8.5</b> Average rlog per sample</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Longitudinal measures of RNA-seq expression and disease activity in FSHD muscle biopsies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="immune_infiltrate" class="section level1">
<h1><span class="header-section-number">Chapter 8</span> Immune Cell Infiltration</h1>
<p>In previous chapters, many genes responsible for immune/inflammotary response showed a trend of elevated expression levels in FSHD samples. This chapter made further analysis on examing the evidence of immune cell infiltration in FSHD affected muscle biopsies and the associtation with disease activities and progression.</p>
<div id="immune-cell-markers" class="section level2">
<h2><span class="header-section-number">8.1</span> Immune cell markers</h2>
<p>The immune cell markers we considered here are annotated by Nano String. There are total 24 cell types with 105 genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">immune_cell &lt;-<span class="st"> </span><span class="kw">read.delim</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;immune_cell_markers&quot;</span>, 
                                    <span class="st">&quot;nanoStringImmuneCellMarkers.txt&quot;</span>))
immune_cell &lt;-<span class="st"> </span>immune_cell <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(gene <span class="op">%in%</span><span class="st"> </span><span class="kw">rowData</span>(sanitized.dds)<span class="op">$</span>gene_name) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">gencode_id =</span> <span class="kw">get_ensembl</span>(<span class="kw">as.character</span>(gene), sanitized.dds))</code></pre></div>
<pre><code>## EBI3 IDO1 LAMP3 OAS3 BLK CD19 CR2 HLA-DOB MS4A1 TNFRSF17 CD8A CD8B FLT3LG GZMM PRF1 GNLY GZMA GZMH KLRD1 KLRF1 CCL13 CCL17 CCL22 CD209 HSD11B1 CCR3 IL5RA PTGDR2 SMPD3 THBS1 CD1A CD1B CD1E F13A1 SYT17 APOE CD68 CHIT1 CXCL5 MARCO MSR1 CMA1 CTSG KIT MS4A2 PRG2 TPSAB1 CSF3R FPR2 MME FOXJ1 MPPED1 PLA2G6 RRAD GTF3C1 GZMB IL21R BCL2 NCR1 ZNF205 IL3RA ANP32B BATF NUP107 CD28 ICOS CD2 CD3E CD3G CD6 ATM DOCK9 NEFL REPS1 USP9Y AKT3 CCR2 EWSR1 LTK NFATC4 BCL6 CXCL13 MAF PDCD1 CD160 FEZ1 CD38 CSF2 IFNG IL12RB2 LTA STAT4 TBX21 CTLA4 IL17A IL17RA RORC CXCR6 GATA3 IL26 LAIR2 PMCH SMAD2 STAT6 FOXP3</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#&#39; list the genes</span>
tmp &lt;-<span class="st"> </span>immune_cell <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">group_by</span>(cellType) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">summarise</span>(<span class="dt">markers=</span><span class="kw">paste</span>(<span class="kw">as.character</span>(gene), <span class="dt">collapse=</span><span class="st">&quot;, &quot;</span>))
knitr<span class="op">::</span><span class="kw">kable</span>(tmp, <span class="dt">caption=</span><span class="st">&quot;Immune cell makers by cell types.&quot;</span>)</code></pre></div>
<table>
<caption>(#tab:load_immune_cell_info)Immune cell makers by cell types.</caption>
<thead>
<tr class="header">
<th align="left">cellType</th>
<th align="left">markers</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">aDC</td>
<td align="left">EBI3, IDO1, LAMP3, OAS3</td>
</tr>
<tr class="even">
<td align="left">B.cell</td>
<td align="left">BLK, CD19, CR2, HLA-DOB, MS4A1, TNFRSF17</td>
</tr>
<tr class="odd">
<td align="left">CD8.T.cell</td>
<td align="left">CD8A, CD8B, FLT3LG, GZMM, PRF1</td>
</tr>
<tr class="even">
<td align="left">Cytotoxic.cell</td>
<td align="left">GNLY, GZMA, GZMH, KLRD1, KLRF1</td>
</tr>
<tr class="odd">
<td align="left">DC</td>
<td align="left">CCL13, CCL17, CCL22, CD209, HSD11B1</td>
</tr>
<tr class="even">
<td align="left">Eosinophils</td>
<td align="left">CCR3, IL5RA, PTGDR2, SMPD3, THBS1</td>
</tr>
<tr class="odd">
<td align="left">iDC</td>
<td align="left">CD1A, CD1B, CD1E, F13A1, SYT17</td>
</tr>
<tr class="even">
<td align="left">Macrophages</td>
<td align="left">APOE, CD68, CHIT1, CXCL5, MARCO, MSR1</td>
</tr>
<tr class="odd">
<td align="left">Mast.cell</td>
<td align="left">CMA1, CTSG, KIT, MS4A2, PRG2, TPSAB1</td>
</tr>
<tr class="even">
<td align="left">Neutrophils</td>
<td align="left">CSF3R, FPR2, MME</td>
</tr>
<tr class="odd">
<td align="left">NK.CD56bright.cell</td>
<td align="left">FOXJ1, MPPED1, PLA2G6, RRAD</td>
</tr>
<tr class="even">
<td align="left">NK.CD56dim.cell</td>
<td align="left">GTF3C1, GZMB, IL21R</td>
</tr>
<tr class="odd">
<td align="left">NK.cell</td>
<td align="left">BCL2, NCR1, ZNF205</td>
</tr>
<tr class="even">
<td align="left">pDC</td>
<td align="left">IL3RA</td>
</tr>
<tr class="odd">
<td align="left">T.cell</td>
<td align="left">CD2, CD3E, CD3G, CD6</td>
</tr>
<tr class="even">
<td align="left">T.helper.cell</td>
<td align="left">ANP32B, BATF, NUP107, CD28, ICOS</td>
</tr>
<tr class="odd">
<td align="left">Tcm</td>
<td align="left">ATM, DOCK9, NEFL, REPS1, USP9Y</td>
</tr>
<tr class="even">
<td align="left">Tem</td>
<td align="left">AKT3, CCR2, EWSR1, LTK, NFATC4</td>
</tr>
<tr class="odd">
<td align="left">TFH</td>
<td align="left">BCL6, CXCL13, MAF, PDCD1</td>
</tr>
<tr class="even">
<td align="left">Tgd</td>
<td align="left">CD160, FEZ1</td>
</tr>
<tr class="odd">
<td align="left">Th1.cell</td>
<td align="left">CD38, CSF2, IFNG, IL12RB2, LTA, STAT4, TBX21, CTLA4</td>
</tr>
<tr class="even">
<td align="left">Th17.cell</td>
<td align="left">IL17A, IL17RA, RORC</td>
</tr>
<tr class="odd">
<td align="left">Th2.cell</td>
<td align="left">CXCR6, GATA3, IL26, LAIR2, PMCH, SMAD2, STAT6</td>
</tr>
<tr class="even">
<td align="left">Treg</td>
<td align="left">FOXP3</td>
</tr>
</tbody>
</table>
</div>
<div id="immune-cell-type-scores-by-tpm-per-samples" class="section level2">
<h2><span class="header-section-number">8.2</span> Immune cell type scores by TPM per Samples</h2>
<p>For each sample, 24 immune cell type scores are given by average <span class="math inline">\(\log_{10}(TPM+1)\)</span> of associated markers. Boxplot below presents immune cell type scores of each samples, seperated by FSHD classes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lv &lt;-<span class="st"> </span><span class="kw">levels</span>(immune_cell<span class="op">$</span>cellType)
immune_cell_tpm &lt;-<span class="st"> </span><span class="kw">sapply</span>(lv, <span class="cf">function</span>(cell_type) {
   immune_id &lt;-<span class="st"> </span>immune_cell <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(cellType <span class="op">==</span><span class="st"> </span>cell_type) <span class="op">%&gt;%</span>
<span class="st">      </span><span class="kw">pull</span>(gencode_id)
   sub &lt;-<span class="st"> </span>sanitized.dds[immune_id]
   tpm_score &lt;-<span class="st"> </span><span class="kw">colMeans</span>(<span class="kw">log10</span>(<span class="kw">assays</span>(sub)[[<span class="st">&quot;TPM&quot;</span>]]<span class="op">+</span><span class="dv">1</span>))
})

immune_cell_tpm &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(immune_cell_tpm) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var=</span><span class="st">&quot;sample_name&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">add_column</span>(<span class="dt">cluster =</span> sanitized.dds<span class="op">$</span>new_cluster_name, 
             <span class="dt">cluster_color =</span>sanitized.dds<span class="op">$</span>cluster_color) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">gather</span>(<span class="dt">key=</span>cell_type, <span class="dt">value=</span>mean_tpm, <span class="op">-</span>sample_name, <span class="op">-</span>cluster, <span class="op">-</span>cluster_color) 

cluster_tpm_score &lt;-<span class="st"> </span>immune_cell_tpm <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">group_by</span>(cell_type, cluster) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">summarise</span>(<span class="dt">cell_type_per_cluster=</span><span class="kw">mean</span>(mean_tpm))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">color_manual &lt;-<span class="st"> </span>immune_cell_tpm <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(cluster) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">summarise</span>(<span class="dt">color=</span><span class="kw">unique</span>(cluster_color)) 

<span class="kw">ggplot</span>(immune_cell_tpm, <span class="kw">aes</span>(<span class="dt">x=</span>cluster, <span class="dt">y=</span>mean_tpm, <span class="dt">color=</span>cluster)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_boxplot</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="st"> </span>cell_type, <span class="dt">scale=</span><span class="st">&quot;free&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(), <span class="dt">legend.justification=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">0</span>), <span class="dt">legend.position=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">0</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>color_manual<span class="op">$</span>color) <span class="op">+</span>
<span class="st">  </span><span class="kw">labs</span>(<span class="dt">y=</span><span class="st">&quot;log10(TPM +1)&quot;</span>, <span class="dt">x=</span><span class="st">&quot;FSHD classes&quot;</span>, <span class="dt">title=</span><span class="st">&quot;Immune cell infiltrate score by TPM&quot;</span>)</code></pre></div>
<p><img src="gitbook_files/figure-html/vis_infiltrate-1.png" width="960" /></p>
</div>
<div id="immune-cell-marker-scores-by-logfc-per-fshd-classes" class="section level2">
<h2><span class="header-section-number">8.3</span> Immune cell marker scores by logFC per FSHD classes</h2>
<p>For each immune cell markers, scores are given as the logFC for each FSHD classes (Mild, Moderate, IG-High, High and Muscle-Low compared to the controls).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">names &lt;-<span class="st"> </span><span class="kw">resultsNames</span>(deseq2_design_cluster)[<span class="op">-</span><span class="dv">1</span>]
immune_marker_logFC &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">resultsNames</span>(deseq2_design_cluster)[<span class="op">-</span><span class="dv">1</span>], <span class="cf">function</span>(results_name){
  res &lt;-<span class="st"> </span><span class="kw">results</span>(deseq2_design_cluster, <span class="dt">name=</span>results_name)
  res[<span class="kw">as.character</span>(immune_cell<span class="op">$</span>gencode_id), <span class="st">&quot;log2FoldChange&quot;</span>, drop=<span class="ot">FALSE</span>]
})
<span class="kw">names</span>(immune_marker_logFC) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Moderate&quot;</span>, <span class="st">&quot;IG-High&quot;</span>, <span class="st">&quot;High&quot;</span>, <span class="st">&quot;Muscle-Low&quot;</span>)
immune_marker_logFC &lt;-<span class="st"> </span><span class="kw">as</span>(<span class="kw">do.call</span>(cbind, immune_marker_logFC), <span class="st">&quot;data.frame&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var=</span><span class="st">&quot;gencode_id&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">left_join</span>(immune_cell, <span class="dt">by=</span><span class="st">&quot;gencode_id&quot;</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#&#39; boxplot of immune marker per celltype by FSHD classes</span>
tidy_marker_logFC &lt;-<span class="st"> </span>immune_marker_logFC <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">gather</span>(<span class="dt">key=</span>classes, <span class="dt">value=</span>log2FoldChange, <span class="op">-</span>gencode_id, <span class="op">-</span>gene, <span class="op">-</span>altName, <span class="op">-</span>cellType) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">classes=</span><span class="kw">factor</span>(classes, 
                               <span class="dt">levels=</span><span class="kw">paste0</span>(<span class="kw">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Moderate&quot;</span>, <span class="st">&quot;IG-High&quot;</span>,
                                               <span class="st">&quot;High&quot;</span>, <span class="st">&quot;Muscle-Low&quot;</span>), <span class="st">&quot;.log2FoldChange&quot;</span>)))
<span class="kw">ggplot</span>(tidy_marker_logFC, <span class="kw">aes</span>(<span class="dt">x=</span>classes, <span class="dt">y=</span>log2FoldChange, <span class="dt">color=</span>classes)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">outlier.shape=</span><span class="ot">NA</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_jitter</span>(<span class="dt">size=</span><span class="fl">0.5</span>, <span class="dt">width=</span><span class="fl">0.5</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>cellType, <span class="dt">scale=</span><span class="st">&quot;free&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>color_manual<span class="op">$</span>color[<span class="op">-</span><span class="dv">1</span>]) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(), <span class="dt">legend.justification=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">0</span>), <span class="dt">legend.position=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">0</span>)) </code></pre></div>
<p><img src="gitbook_files/figure-html/viz_immune_marker_logFC-1.png" width="960" /></p>
</div>
<div id="frequency-of-significance" class="section level2">
<h2><span class="header-section-number">8.4</span> Frequency of significance</h2>
<p>Code chunk below calculates frequency of significent markers (log2FC &gt; 2) by cell types.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">freq_sig &lt;-<span class="st"> </span>tidy_marker_logFC <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">group_by</span>(cellType, classes) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">summarise</span>(<span class="dt">n_markers=</span><span class="kw">n</span>(),
            <span class="dt">frequency =</span> <span class="kw">sum</span>(log2FoldChange <span class="op">&gt;</span><span class="dv">2</span>) <span class="op">/</span><span class="st"> </span>n_markers) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">label =</span> <span class="kw">paste0</span>(cellType, <span class="st">&quot; (&quot;</span>, n_markers, <span class="st">&quot;)&quot;</span>))
  <span class="co">#summarise(n_markers=nrow(.)/length(levels(classes)), </span>
  <span class="co">#          freq = sum(log2FoldChange &gt; 2) / n_markers)</span>
<span class="kw">ggplot</span>(freq_sig, <span class="kw">aes</span>(<span class="dt">x=</span>classes, <span class="dt">y=</span>frequency, <span class="dt">color=</span>classes, <span class="dt">fill=</span>classes)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="st"> </span>label) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>color_manual<span class="op">$</span>color[<span class="op">-</span><span class="dv">1</span>]) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span>color_manual<span class="op">$</span>color[<span class="op">-</span><span class="dv">1</span>]) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(), <span class="dt">legend.justification=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">0</span>), <span class="dt">legend.position=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">0</span>),
        <span class="dt">legend.key.size=</span><span class="kw">unit</span>(.<span class="dv">5</span>, <span class="st">&quot;cm&quot;</span>)) </code></pre></div>
<p><img src="gitbook_files/figure-html/freq_of_significance-1.png" width="1056" /></p>
</div>
<div id="average-rlog-per-sample" class="section level2">
<h2><span class="header-section-number">8.5</span> Average rlog per sample</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sub &lt;-<span class="st"> </span>sanitized.rlg[<span class="kw">as.character</span>(immune_cell<span class="op">$</span>gencode_id)]
immune_sample_score &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">average=</span><span class="kw">colMeans</span>(<span class="kw">assay</span>(sub))) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var=</span><span class="st">&quot;sample_name&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">add_column</span>(<span class="dt">pheno_type =</span> sub<span class="op">$</span>pheno_type,
             <span class="dt">classes =</span> sub<span class="op">$</span>new_cluster_name)
<span class="kw">ggplot</span>(immune_sample_score, <span class="kw">aes</span>(<span class="dt">x=</span>classes, <span class="dt">y=</span>average, <span class="dt">color=</span>classes)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">outlier.shape=</span><span class="ot">NA</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_jitter</span>(<span class="dt">width=</span><span class="fl">0.3</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>color_manual<span class="op">$</span>color)</code></pre></div>
<p><img src="gitbook_files/figure-html/boxplot_ave_rlog_per_sample-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="pax7.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["gitbook.pdf", "gitbook.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
