<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Classes of RNA-seq biopsy sampels | Longitudinal measures of RNA-seq expression and disease activity in FSHD muscle biopsies</title>
  <meta name="description" content="Bioinformatics and statistics analysis for the logitudinal study of RNA-seq expression and disear acitivity in FSHD muscle bipsies" />
  <meta name="generator" content="bookdown 0.14 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Classes of RNA-seq biopsy sampels | Longitudinal measures of RNA-seq expression and disease activity in FSHD muscle biopsies" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Bioinformatics and statistics analysis for the logitudinal study of RNA-seq expression and disear acitivity in FSHD muscle bipsies" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Classes of RNA-seq biopsy sampels | Longitudinal measures of RNA-seq expression and disease activity in FSHD muscle biopsies" />
  
  <meta name="twitter:description" content="Bioinformatics and statistics analysis for the logitudinal study of RNA-seq expression and disear acitivity in FSHD muscle bipsies" />
  

<meta name="author" content="Chao-Jen Wong" />


<meta name="date" content="2020-02-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="longitudinal-changes.html"/>
<link rel="next" href="mild-fshd.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">RNA-seq data analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#package-structure"><i class="fa fa-check"></i><b>1.1</b> Package structure</a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#datasets"><i class="fa fa-check"></i><b>1.1.1</b> Datasets</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#geo"><i class="fa fa-check"></i><b>1.2</b> GEO</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#system-requirment"><i class="fa fa-check"></i><b>1.3</b> System requirment</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="preprocessing.html"><a href="preprocessing.html"><i class="fa fa-check"></i><b>2</b> Preprocessing and gene counts</a><ul>
<li class="chapter" data-level="2.1" data-path="preprocessing.html"><a href="preprocessing.html#shell-script-for-pre-processing"><i class="fa fa-check"></i><b>2.1</b> Shell script for pre-processing</a></li>
<li class="chapter" data-level="2.2" data-path="preprocessing.html"><a href="preprocessing.html#annotation-txdb-package"><i class="fa fa-check"></i><b>2.2</b> Annotation TxDb package</a></li>
<li class="chapter" data-level="2.3" data-path="preprocessing.html"><a href="preprocessing.html#r-script-for-gene-counts"><i class="fa fa-check"></i><b>2.3</b> R Script for gene counts</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="follow-up-visit.html"><a href="follow-up-visit.html"><i class="fa fa-check"></i><b>3</b> Follow-up visit</a><ul>
<li class="chapter" data-level="3.1" data-path="follow-up-visit.html"><a href="follow-up-visit.html#loading-datasets"><i class="fa fa-check"></i><b>3.1</b> Loading datasets</a></li>
<li class="chapter" data-level="3.2" data-path="follow-up-visit.html"><a href="follow-up-visit.html#the-follow-up-visit-rna-seq-dataset"><i class="fa fa-check"></i><b>3.2</b> The follow-up visit RNA-seq dataset</a></li>
<li class="chapter" data-level="3.3" data-path="follow-up-visit.html"><a href="follow-up-visit.html#dux4-and-other-marker-scores"><i class="fa fa-check"></i><b>3.3</b> DUX4 and other marker scores</a><ul>
<li class="chapter" data-level="3.3.1" data-path="follow-up-visit.html"><a href="follow-up-visit.html#group-samples-by-dux4-score"><i class="fa fa-check"></i><b>3.3.1</b> Group samples by DUX4 score</a></li>
<li class="chapter" data-level="3.3.2" data-path="follow-up-visit.html"><a href="follow-up-visit.html#heatmap-of-the-discriminative-genes"><i class="fa fa-check"></i><b>3.3.2</b> Heatmap of the discriminative genes</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="follow-up-visit.html"><a href="follow-up-visit.html#dux4-candidate-biomarkers-are-eleveated-in-fshd"><i class="fa fa-check"></i><b>3.4</b> DUX4 candidate biomarkers are eleveated in FSHD</a></li>
<li class="chapter" data-level="3.5" data-path="follow-up-visit.html"><a href="follow-up-visit.html#correlation-of-dux4-score-to-mri-and-histopathology"><i class="fa fa-check"></i><b>3.5</b> Correlation of DUX4 score to MRI and histopathology</a><ul>
<li class="chapter" data-level="3.5.1" data-path="follow-up-visit.html"><a href="follow-up-visit.html#dux4-and-pathology-score"><i class="fa fa-check"></i><b>3.5.1</b> DUX4 and pathology score</a></li>
<li class="chapter" data-level="3.5.2" data-path="follow-up-visit.html"><a href="follow-up-visit.html#dux4-and-stir-signal"><i class="fa fa-check"></i><b>3.5.2</b> DUX4 and STIR signal</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="follow-up-visit.html"><a href="follow-up-visit.html#dux4-score-and-t1-fat-fraction"><i class="fa fa-check"></i><b>3.6</b> DUX4 score and T1 fat fraction</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="longitudinal-changes.html"><a href="longitudinal-changes.html"><i class="fa fa-check"></i><b>4</b> Longitudinal changes over two visits</a><ul>
<li class="chapter" data-level="4.1" data-path="longitudinal-changes.html"><a href="longitudinal-changes.html#loading-datasets-1"><i class="fa fa-check"></i><b>4.1</b> Loading datasets</a></li>
<li class="chapter" data-level="4.2" data-path="longitudinal-changes.html"><a href="longitudinal-changes.html#pathology-scores"><i class="fa fa-check"></i><b>4.2</b> Pathology scores</a></li>
<li class="chapter" data-level="4.3" data-path="longitudinal-changes.html"><a href="longitudinal-changes.html#change-of-marker-scores-over-two-visits"><i class="fa fa-check"></i><b>4.3</b> Change of marker scores over two visits</a><ul>
<li class="chapter" data-level="4.3.1" data-path="longitudinal-changes.html"><a href="longitudinal-changes.html#dux4-marker-scores"><i class="fa fa-check"></i><b>4.3.1</b> DUX4 marker scores</a></li>
<li class="chapter" data-level="4.3.2" data-path="longitudinal-changes.html"><a href="longitudinal-changes.html#extracellular-matrix-structure-scores"><i class="fa fa-check"></i><b>4.3.2</b> Extracellular matrix structure scores</a></li>
<li class="chapter" data-level="4.3.3" data-path="longitudinal-changes.html"><a href="longitudinal-changes.html#immuse-responseinflammatory-score"><i class="fa fa-check"></i><b>4.3.3</b> Immuse response/Inflammatory score</a></li>
<li class="chapter" data-level="4.3.4" data-path="longitudinal-changes.html"><a href="longitudinal-changes.html#immunoglobulin-score"><i class="fa fa-check"></i><b>4.3.4</b> Immunoglobulin score</a></li>
<li class="chapter" data-level="4.3.5" data-path="longitudinal-changes.html"><a href="longitudinal-changes.html#cell-cycle-score"><i class="fa fa-check"></i><b>4.3.5</b> Cell cycle score</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="longitudinal-changes.html"><a href="longitudinal-changes.html#pathology-and-dux4-scores-changes"><i class="fa fa-check"></i><b>4.4</b> Pathology and DUX4 scores changes</a></li>
<li class="chapter" data-level="4.5" data-path="longitudinal-changes.html"><a href="longitudinal-changes.html#markers-by-visits-and-stir-ratings"><i class="fa fa-check"></i><b>4.5</b> Markers by visits and STIR ratings</a><ul>
<li class="chapter" data-level="4.5.1" data-path="longitudinal-changes.html"><a href="longitudinal-changes.html#dux4-scores-and-stir-ratings"><i class="fa fa-check"></i><b>4.5.1</b> DUX4 scores and STIR ratings</a></li>
<li class="chapter" data-level="4.5.2" data-path="longitudinal-changes.html"><a href="longitudinal-changes.html#other-markers"><i class="fa fa-check"></i><b>4.5.2</b> Other markers</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="rnaseq-classes.html"><a href="rnaseq-classes.html"><i class="fa fa-check"></i><b>5</b> Classes of RNA-seq biopsy sampels</a><ul>
<li class="chapter" data-level="5.1" data-path="rnaseq-classes.html"><a href="rnaseq-classes.html#fshd-classes"><i class="fa fa-check"></i><b>5.1</b> FSHD classes</a><ul>
<li class="chapter" data-level="5.1.1" data-path="rnaseq-classes.html"><a href="rnaseq-classes.html#load-datasets-and-libraries"><i class="fa fa-check"></i><b>5.1.1</b> Load datasets and libraries</a></li>
<li class="chapter" data-level="5.1.2" data-path="rnaseq-classes.html"><a href="rnaseq-classes.html#feature-selection"><i class="fa fa-check"></i><b>5.1.2</b> Feature selection</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="rnaseq-classes.html"><a href="rnaseq-classes.html#sample-space-and-k-means-clustering"><i class="fa fa-check"></i><b>5.2</b> Sample space and k-means clustering</a></li>
<li class="chapter" data-level="5.3" data-path="rnaseq-classes.html"><a href="rnaseq-classes.html#visualize-the-fshd-classes"><i class="fa fa-check"></i><b>5.3</b> Visualize the FSHD classes</a></li>
<li class="chapter" data-level="5.4" data-path="rnaseq-classes.html"><a href="rnaseq-classes.html#view-by-selected-marker-genes"><i class="fa fa-check"></i><b>5.4</b> View by selected marker genes</a></li>
<li class="chapter" data-level="5.5" data-path="rnaseq-classes.html"><a href="rnaseq-classes.html#mri-characterisitcs-and-pathology-scores-by-fshd-classes"><i class="fa fa-check"></i><b>5.5</b> MRI characterisitcs and pathology scores by FSHD classes</a></li>
<li class="chapter" data-level="5.6" data-path="rnaseq-classes.html"><a href="rnaseq-classes.html#discriminative-biomarkers-characteristics-by-fshd-classes"><i class="fa fa-check"></i><b>5.6</b> Discriminative biomarkers characteristics by FSHD classes</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="mild-fshd.html"><a href="mild-fshd.html"><i class="fa fa-check"></i><b>6</b> Markers discriminate Mild FSHD from control samples</a><ul>
<li class="chapter" data-level="6.1" data-path="mild-fshd.html"><a href="mild-fshd.html#differentially-up-regulated-in-mild-fshd-samples"><i class="fa fa-check"></i><b>6.1</b> Differentially up-regulated in Mild FSHD samples</a></li>
<li class="chapter" data-level="6.2" data-path="mild-fshd.html"><a href="mild-fshd.html#select-fshd-relavent-candidates"><i class="fa fa-check"></i><b>6.2</b> Select FSHD relavent candidates</a></li>
<li class="chapter" data-level="6.3" data-path="mild-fshd.html"><a href="mild-fshd.html#go-analysis"><i class="fa fa-check"></i><b>6.3</b> GO analysis</a></li>
<li class="chapter" data-level="6.4" data-path="mild-fshd.html"><a href="mild-fshd.html#cross-validation"><i class="fa fa-check"></i><b>6.4</b> Cross validation</a></li>
<li class="chapter" data-level="6.5" data-path="mild-fshd.html"><a href="mild-fshd.html#discriminative-power-per-candidates"><i class="fa fa-check"></i><b>6.5</b> Discriminative power per candidates</a></li>
<li class="chapter" data-level="6.6" data-path="mild-fshd.html"><a href="mild-fshd.html#heatmap-of-some-candidates"><i class="fa fa-check"></i><b>6.6</b> Heatmap of some candidates</a></li>
<li class="chapter" data-level="6.7" data-path="mild-fshd.html"><a href="mild-fshd.html#some-candidates"><i class="fa fa-check"></i><b>6.7</b> Some candidates</a><ul>
<li class="chapter" data-level="6.7.1" data-path="mild-fshd.html"><a href="mild-fshd.html#ckdn1a"><i class="fa fa-check"></i><b>6.7.1</b> CKDN1A</a></li>
<li class="chapter" data-level="6.7.2" data-path="mild-fshd.html"><a href="mild-fshd.html#ccna1"><i class="fa fa-check"></i><b>6.7.2</b> CCNA1</a></li>
<li class="chapter" data-level="6.7.3" data-path="mild-fshd.html"><a href="mild-fshd.html#a-few-more-examples"><i class="fa fa-check"></i><b>6.7.3</b> A few more examples</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="pax7.html"><a href="pax7.html"><i class="fa fa-check"></i><b>7</b> PAX7 target mis-regulation in FSHD biopsy RNA-seq samples</a><ul>
<li class="chapter" data-level="7.1" data-path="pax7.html"><a href="pax7.html#pa7-scores-and-disease-activity"><i class="fa fa-check"></i><b>7.1</b> PA7 scores and disease activity</a><ul>
<li class="chapter" data-level="7.1.1" data-path="pax7.html"><a href="pax7.html#boxplot-of-pax7-score"><i class="fa fa-check"></i><b>7.1.1</b> Boxplot of PAX7 score</a></li>
<li class="chapter" data-level="7.1.2" data-path="pax7.html"><a href="pax7.html#heapmap-of-pax7-targeted-gene-expression"><i class="fa fa-check"></i><b>7.1.2</b> Heapmap of PAX7 targeted gene expression</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="pax7.html"><a href="pax7.html#view-from-a-different-angle"><i class="fa fa-check"></i><b>7.2</b> View from a different angle</a></li>
<li class="chapter" data-level="7.3" data-path="pax7.html"><a href="pax7.html#the-paradox"><i class="fa fa-check"></i><b>7.3</b> The paradox</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html"><i class="fa fa-check"></i><b>8</b> Immune Cell Infiltration</a><ul>
<li class="chapter" data-level="8.1" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html#immune-cell-markers"><i class="fa fa-check"></i><b>8.1</b> Immune cell markers</a></li>
<li class="chapter" data-level="8.2" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html#immune-cell-type-scores-by-tpm-per-samples"><i class="fa fa-check"></i><b>8.2</b> Immune cell type scores by TPM per Samples</a></li>
<li class="chapter" data-level="8.3" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html#immune-cell-marker-scores-by-logfc-per-fshd-classes"><i class="fa fa-check"></i><b>8.3</b> Immune cell marker scores by logFC per FSHD classes</a></li>
<li class="chapter" data-level="8.4" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html#frequency-of-significance"><i class="fa fa-check"></i><b>8.4</b> Frequency of significance</a></li>
<li class="chapter" data-level="8.5" data-path="immune-infiltrate.html"><a href="immune-infiltrate.html#average-rlog-per-sample"><i class="fa fa-check"></i><b>8.5</b> Average rlog per sample</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="biopsy-qc.html"><a href="biopsy-qc.html"><i class="fa fa-check"></i><b>9</b> Biopsy Quality Control</a><ul>
<li class="chapter" data-level="9.1" data-path="biopsy-qc.html"><a href="biopsy-qc.html#blood-fat-and-muscle-cell-markers"><i class="fa fa-check"></i><b>9.1</b> Blood, fat and muscle cell markers</a></li>
<li class="chapter" data-level="9.2" data-path="biopsy-qc.html"><a href="biopsy-qc.html#define-score"><i class="fa fa-check"></i><b>9.2</b> Define score</a></li>
<li class="chapter" data-level="9.3" data-path="biopsy-qc.html"><a href="biopsy-qc.html#visualization"><i class="fa fa-check"></i><b>9.3</b> Visualization</a></li>
<li class="chapter" data-level="9.4" data-path="biopsy-qc.html"><a href="biopsy-qc.html#determine-outliers"><i class="fa fa-check"></i><b>9.4</b> Determine “outliers”</a></li>
<li class="chapter" data-level="9.5" data-path="biopsy-qc.html"><a href="biopsy-qc.html#determine-bad-biopsies"><i class="fa fa-check"></i><b>9.5</b> Determine “bad” biopsies?</a></li>
<li class="chapter" data-level="9.6" data-path="biopsy-qc.html"><a href="biopsy-qc.html#sample-01-0037"><i class="fa fa-check"></i><b>9.6</b> Sample 01-0037</a></li>
<li class="chapter" data-level="9.7" data-path="biopsy-qc.html"><a href="biopsy-qc.html#tables"><i class="fa fa-check"></i><b>9.7</b> Tables</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Longitudinal measures of RNA-seq expression and disease activity in FSHD muscle biopsies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="rnaseq-classes" class="section level1">
<h1><span class="header-section-number">Chapter 5</span> Classes of RNA-seq biopsy sampels</h1>
<p>To better understand the differences between different FSHD samples and identify sub-groups and their characteristics, we combine the RNA-seq gene expression matrics from the first and follow-up visit samples and apply k-means clustering on the sample space build upon selected features. We are able to find five anchor samples and partition all FSHD and controls into five clusters, each of which has unique characteristics.</p>
<div id="fshd-classes" class="section level2">
<h2><span class="header-section-number">5.1</span> FSHD classes</h2>
<p>Steps below explained how the samples are paritioned into five clusters:<br />
1. Feature selection: select features that are best characterize the cluster patterns of PCA sample space built on all samples. We end up select the differentially expressed genes in the samples characterized by high expressed DUX4-regulated and inflammatory response genes (<code>cc_sample</code>), as well as muscle development related genes. The feature space is reduced to 705 informative features (Supple. Table S4 and Suppl. Fig. S5B).<br />
2. Sample space construction: use previous selected featrues and their regularized log expression to build sample sapce by PCA. The dimension is then reduced. 3. K-means clusterings: based on the PCA sample space and k-mean cluster is used to partition all samples into five clusters. To enchance the stability of k-means algorithm, we select five anchor samples (<code>01-0041</code>, <code>01-0022-1</code>, <code>01-0029b</code>,<code>32-0002b1</code>, <code>32-0012b</code>) representing the center of each cluster.</p>
<div id="load-datasets-and-libraries" class="section level3">
<h3><span class="header-section-number">5.1.1</span> Load datasets and libraries</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># load datasets and libraries</span>
<span class="kw">library</span>(BiocParallel)
multi_param &lt;-<span class="st"> </span><span class="kw">MulticoreParam</span>(<span class="dt">worker=</span><span class="dv">2</span>)
<span class="kw">register</span>(multi_param, <span class="dt">default=</span><span class="ot">TRUE</span>)

<span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(DESeq2))
<span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(xlsx))
<span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(ggplot2))
<span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(tidyverse))
<span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(pheatmap))
<span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(ggrepel))
<span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(gridExtra))

pkg_dir &lt;-<span class="st"> &quot;/fh/fast/tapscott_s/CompBio/RNA-Seq/hg38.FSHD.biopsy.all&quot;</span>
<span class="kw">source</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;scripts&quot;</span>, <span class="st">&quot;manuscript_tools.R&quot;</span>))
<span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;public_data&quot;</span>, <span class="st">&quot;mri_pathology.rda&quot;</span>))
<span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;public_data&quot;</span>, <span class="st">&quot;sanitized.dds.rda&quot;</span>))
<span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;public_data&quot;</span>, <span class="st">&quot;sanitized.rlg.rda&quot;</span>))
<span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;public_data&quot;</span>, <span class="st">&quot;cc_sample.rda&quot;</span>))
<span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;public_data&quot;</span>, <span class="st">&quot;mri_pathology.rda&quot;</span>))</code></pre></div>
</div>
<div id="feature-selection" class="section level3">
<h3><span class="header-section-number">5.1.2</span> Feature selection</h3>
<p>The FSHD samples that exhibit strong DUX4 and inflammatory signatures are</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ccp &lt;-<span class="st"> </span>cc_sample <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(DUX4_group <span class="op">==</span><span class="st"> &quot;DUX4+&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">filter</span>(cc_intensity <span class="op">!=</span><span class="st"> &quot;cc++&quot;</span>)
knitr<span class="op">::</span><span class="kw">kable</span>(ccp, <span class="dt">caption=</span><span class="st">&quot;Samples showing strong DUX4 and inflammatory signatures.&quot;</span>)</code></pre></div>
<table>
<caption>(#tab:select_features_1)Samples showing strong DUX4 and inflammatory signatures.</caption>
<thead>
<tr class="header">
<th align="left">sample_name</th>
<th align="left">cc_group</th>
<th align="left">DUX4_group</th>
<th align="left">pheno_type</th>
<th align="left">visit</th>
<th align="left">cc_intensity</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">32-0005b</td>
<td align="left">cc+</td>
<td align="left">DUX4+</td>
<td align="left">FSHD</td>
<td align="left">II</td>
<td align="left">cc+</td>
</tr>
<tr class="even">
<td align="left">32-0002b1</td>
<td align="left">cc+</td>
<td align="left">DUX4+</td>
<td align="left">FSHD</td>
<td align="left">II</td>
<td align="left">cc+</td>
</tr>
<tr class="odd">
<td align="left">32-0003</td>
<td align="left">cc+</td>
<td align="left">DUX4+</td>
<td align="left">FSHD</td>
<td align="left">I</td>
<td align="left">cc+</td>
</tr>
<tr class="even">
<td align="left">01-0029</td>
<td align="left">cc+</td>
<td align="left">DUX4+</td>
<td align="left">FSHD</td>
<td align="left">I</td>
<td align="left">cc+</td>
</tr>
<tr class="odd">
<td align="left">32-0005</td>
<td align="left">cc+</td>
<td align="left">DUX4+</td>
<td align="left">FSHD</td>
<td align="left">I</td>
<td align="left">cc+</td>
</tr>
<tr class="even">
<td align="left">32-0004</td>
<td align="left">cc+</td>
<td align="left">DUX4+</td>
<td align="left">FSHD</td>
<td align="left">I</td>
<td align="left">cc+</td>
</tr>
<tr class="odd">
<td align="left">32-0006</td>
<td align="left">cc+</td>
<td align="left">DUX4+</td>
<td align="left">FSHD</td>
<td align="left">I</td>
<td align="left">cc+</td>
</tr>
<tr class="even">
<td align="left">01-0025</td>
<td align="left">cc+</td>
<td align="left">DUX4+</td>
<td align="left">FSHD</td>
<td align="left">I</td>
<td align="left">cc+</td>
</tr>
</tbody>
</table>
<p>Select the up-reguated differentially expressed genes in DUX+/inflammed+ samples relative to the controls (adjusted <span class="math inline">\(p\)</span>-vlaue &lt; 0.05 corresponding to <span class="math inline">\(H_0: |lfc| = 2\)</span>). This set of genes are informative in sample variation.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">keep &lt;-<span class="st"> </span><span class="kw">colnames</span>(sanitized.dds) <span class="op">%in%</span><span class="st"> </span>ccp<span class="op">$</span>sample_name <span class="op">|</span>
<span class="st">  </span>sanitized.dds<span class="op">$</span>pheno_type <span class="op">==</span><span class="st"> &quot;Control&quot;</span>
dux4_dds &lt;-<span class="st"> </span>sanitized.dds[, keep]
<span class="kw">design</span>(dux4_dds) &lt;-<span class="st"> </span><span class="er">~</span><span class="st"> </span>pheno_type
dux4_dds &lt;-<span class="st"> </span><span class="kw">DESeq</span>(dux4_dds)
dux4_res &lt;-<span class="st"> </span><span class="kw">results</span>(dux4_dds, <span class="dt">alpha=</span><span class="fl">0.05</span>, <span class="dt">lfcThreshold=</span><span class="dv">2</span>)

<span class="co">#&#39; tidy up the up-regulated</span>
df_up &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(dux4_res) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var=</span><span class="st">&quot;gencode_id&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>log2FoldChange <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">gene_name =</span> <span class="kw">rowData</span>(dux4_dds[gencode_id])<span class="op">$</span>gene_name) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">gene_id =</span>
           <span class="kw">sapply</span>(<span class="kw">strsplit</span>(gencode_id, <span class="st">&quot;.&quot;</span>, <span class="dt">fixed=</span><span class="ot">TRUE</span>), <span class="st">&quot;[[&quot;</span>, <span class="dv">1</span>))</code></pre></div>
<p>Another set of genes that are also informative in terms of sample variation is the muscle development related genes (code below). We hereby select these two set of genes as our most informative features to construct sample space.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># muscle related, mostly differentially down-regulated in the biopsies with less muscle content</span>
muscle &lt;-<span class="st"> </span><span class="kw">c</span>(
  <span class="st">&quot;MYF5&quot;</span>, <span class="st">&quot;MYF6&quot;</span>, <span class="st">&quot;MYH1&quot;</span>, <span class="co"># newly added</span>
  <span class="st">&quot;MYH13&quot;</span>, <span class="st">&quot;MYH2&quot;</span>, <span class="st">&quot;MYH4&quot;</span>, <span class="st">&quot;MYH6&quot;</span>, <span class="st">&quot;MYH7&quot;</span>, <span class="co"># newly added</span>
  <span class="st">&quot;MYHAS&quot;</span>, <span class="st">&quot;MYL1&quot;</span>, <span class="st">&quot;MYL2&quot;</span>, <span class="co">#newly added</span>
  <span class="st">&quot;MYO18B&quot;</span>, <span class="st">&quot;MYOD1&quot;</span>, <span class="st">&quot;MYOG&quot;</span>, <span class="st">&quot;MYOM2&quot;</span>, <span class="st">&quot;MYOM3&quot;</span>,
  <span class="st">&quot;MYOT&quot;</span>, <span class="st">&quot;MYOZ1&quot;</span>, <span class="st">&quot;MYOZ2&quot;</span>, <span class="st">&quot;MYOZ3&quot;</span>, <span class="st">&quot;PAX7&quot;</span>, <span class="st">&quot;PAX3&quot;</span>)
de_markers &lt;-<span class="st"> </span><span class="kw">c</span>(df_up<span class="op">$</span>gencode_id, <span class="kw">unname</span>(<span class="kw">get_ensembl</span>(muscle, sanitized.rlg)))</code></pre></div>
</div>
</div>
<div id="sample-space-and-k-means-clustering" class="section level2">
<h2><span class="header-section-number">5.2</span> Sample space and k-means clustering</h2>
<p>Defining anchor samples helps to stablize the k-means clustering.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#&#39; anchar_sample: 01-0041 (control), 32-0002, &quot;01-0029b&quot;, &quot;32-0002b1&quot;, and &quot;32-0012b&quot;</span>
anchor_sample &lt;-<span class="st"> </span>
<span class="st">  </span><span class="kw">data.frame</span>(<span class="dt">label =</span> <span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;D&quot;</span>, <span class="st">&quot;E&quot;</span>),
             <span class="dt">sample_name =</span> <span class="kw">c</span>(<span class="st">&quot;01-0041&quot;</span>, <span class="st">&quot;01-0022-1&quot;</span>, <span class="st">&quot;01-0029b&quot;</span>,<span class="st">&quot;32-0002b1&quot;</span>, <span class="st">&quot;32-0012b&quot;</span>),
             <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)</code></pre></div>
<p>Build sample space by PCA and call up k-means algorithm.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#&#39; (3b) k-means (k=5) on PCA of de_markers</span>
k &lt;-<span class="st"> </span><span class="dv">5</span>
data &lt;-<span class="st"> </span><span class="kw">assay</span>(sanitized.rlg[de_markers, ])
pca &lt;-<span class="st"> </span><span class="kw">prcomp</span>(<span class="kw">t</span>(data))
<span class="co">#cl &lt;- kmeans(pca$x, centers=k, iter.max=15, nstart=5)</span>
cl &lt;-<span class="st"> </span><span class="kw">kmeans</span>(pca<span class="op">$</span>x, <span class="dt">centers=</span>pca<span class="op">$</span>x[anchor_sample<span class="op">$</span>sample_name, ], <span class="dt">iter.max=</span><span class="dv">15</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">.rename_cluster_<span class="dv">5</span> &lt;-<span class="st"> </span><span class="cf">function</span>(sample_name) {
   anchor_samples &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">A =</span> <span class="st">&quot;01-0041&quot;</span>, <span class="dt">B =</span> <span class="st">&quot;32-0002&quot;</span>, <span class="dt">C =</span> <span class="st">&quot;01-0029b&quot;</span>,
                       <span class="dt">D =</span> <span class="st">&quot;32-0002b1&quot;</span>, <span class="dt">E =</span> <span class="st">&quot;32-0012b&quot;</span>)
   i &lt;-<span class="st"> </span><span class="kw">which</span>(anchor_sample<span class="op">$</span>sample_name <span class="op">%in%</span><span class="st"> </span><span class="kw">as.character</span>(sample_name))
   <span class="kw">names</span>(anchor_samples)[i]
}

rename_cluster &lt;-<span class="st"> </span><span class="cf">function</span>(cl, k) {
  df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">sample_name=</span><span class="kw">names</span>(cl<span class="op">$</span>cluster),
                   <span class="dt">cluster=</span>cl<span class="op">$</span>cluster)
  tmp &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(cluster) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">rename_cluster=</span><span class="kw">.rename_cluster_5</span>(sample_name)) 
  df &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">renamed=</span>tmp<span class="op">$</span>rename_cluster[cluster])
}

.face_off_cluster &lt;-<span class="st"> </span><span class="cf">function</span>(old_name) {
  new_name &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">A=</span><span class="st">&quot;Mild&quot;</span>, <span class="dt">B=</span><span class="st">&quot;Moderate&quot;</span>, <span class="dt">C=</span><span class="st">&quot;IG-High&quot;</span>, <span class="dt">D=</span><span class="st">&quot;High&quot;</span>, <span class="dt">E=</span><span class="st">&quot;Muscle-Low&quot;</span>)
  <span class="kw">factor</span>(new_name[old_name], <span class="dt">levels=</span>new_name)
}

cluster_color &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">A_Cntr=</span><span class="st">&quot;#ff7f00&quot;</span>, <span class="dt">A=</span><span class="st">&quot;#a65628&quot;</span>, <span class="dt">B=</span><span class="st">&quot;#f781bf&quot;</span>, <span class="dt">C=</span><span class="st">&quot;#984ea3&quot;</span>, <span class="dt">D=</span><span class="st">&quot;#e41a1c&quot;</span>, <span class="dt">E=</span><span class="st">&quot;#377eb8&quot;</span>)
cluster_df &lt;-<span class="st"> </span><span class="kw">rename_cluster</span>(cl, k) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">RNA_cluster =</span> <span class="kw">ifelse</span>(sanitized.rlg<span class="op">$</span>pheno_type <span class="op">==</span><span class="st"> &quot;Control&quot;</span>,
                              <span class="kw">paste0</span>(renamed, <span class="st">&quot;_Cntr&quot;</span>), renamed)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">RNA_cluster =</span> <span class="kw">factor</span>(RNA_cluster, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;A_Cntr&quot;</span>, <span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;D&quot;</span>, <span class="st">&quot;E&quot;</span>))) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">color =</span> cluster_color[<span class="kw">as.character</span>(RNA_cluster)])  <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">new_cluster_name =</span> <span class="kw">.face_off_cluster</span>(renamed)) 

sanitized.rlg<span class="op">$</span>RNA_cluster_<span class="dv">5</span> &lt;-<span class="st"> </span>sanitized.dds<span class="op">$</span>RNA_cluster_<span class="dv">5</span> &lt;-<span class="st"> </span>cluster_df<span class="op">$</span>RNA_cluster</code></pre></div>
</div>
<div id="visualize-the-fshd-classes" class="section level2">
<h2><span class="header-section-number">5.3</span> Visualize the FSHD classes</h2>
<p>Showing below is the visualization of FSHD sample space reduced to two dimensions by PCA’s first two components. Colors represent classes clustered by k-means. Cluster A are renamed to Mild, B to Moderate, C to Immunoglobulin-High (IG-High), D to High and E to Muscle-Low.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data &lt;-<span class="st"> </span><span class="kw">plotPCA</span>(sanitized.rlg, 
                <span class="dt">intgroup=</span><span class="kw">c</span>(<span class="st">&quot;RNA_cluster_5&quot;</span>, <span class="st">&quot;pheno_type&quot;</span>), <span class="dt">returnData=</span><span class="ot">TRUE</span>)
percentVar &lt;-<span class="st"> </span><span class="kw">round</span>(<span class="dv">100</span> <span class="op">*</span><span class="st"> </span><span class="kw">attr</span>(data, <span class="st">&quot;percentVar&quot;</span>))
gg &lt;-<span class="st"> </span><span class="kw">ggplot</span>(data, <span class="kw">aes</span>(PC1, PC2, <span class="dt">color=</span>RNA_cluster_<span class="dv">5</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="fl">0.8</span>) <span class="op">+</span>
<span class="st">  </span><span class="co">#xlab(paste0(&quot;PC1: &quot;,percentVar[1],&quot;% variance&quot;)) +</span>
<span class="st">  </span><span class="co">#ylab(paste0(&quot;PC2: &quot;,percentVar[2],&quot;% variance&quot;)) +</span>
<span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;Principal components on initial and follow-up visits&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_text_repel</span>(<span class="kw">aes</span>(<span class="dt">label=</span><span class="kw">rownames</span>(data)), <span class="dt">size=</span><span class="fl">2.2</span>, <span class="dt">show.legend=</span><span class="ot">FALSE</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme_bw</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">name=</span><span class="st">&quot;Classes&quot;</span>, <span class="dt">values=</span>cluster_color) <span class="op">+</span>
<span class="st">  </span><span class="co">#scale_color_brewer(palette=&quot;Set2&quot;) +</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;right&quot;</span>, 
        <span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="fl">0.5</span>, <span class="dt">size=</span><span class="dv">10</span>)) 
gg</code></pre></div>
<p><img src="gitbook_files/figure-html/pca_fshd_classes-1.png" width="672" /></p>
</div>
<div id="view-by-selected-marker-genes" class="section level2">
<h2><span class="header-section-number">5.4</span> View by selected marker genes</h2>
<p>We selected informative biomarkers of six FSHD relevant biological functions and illustrate the characteristics of FSHD classes by heatmap. Row gaps separated by (top to bottom) DUX4-regulated, extracellular matrix, cell cycle, immune/inflammatory response, immunoglobulin and muscle development related. Column gaps separate RNA-seq clustered groups, A-E. Color scale corresponds to row z-score of regularized log2 expression.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">robust_markers &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">dux4 =</span> <span class="kw">c</span>(<span class="st">&quot;ZSCAN4&quot;</span>, <span class="st">&quot;KHDC1L&quot;</span>, <span class="st">&quot;LEUTX&quot;</span>, 
                          <span class="st">&quot;PRAMEF2&quot;</span>, <span class="st">&quot;TRIM43&quot;</span>),
                 <span class="dt">cellular_matrix=</span><span class="kw">c</span>(<span class="st">&quot;PLA2G2A&quot;</span>, <span class="st">&quot;COL19A1&quot;</span>, <span class="st">&quot;COL1A1&quot;</span>,
                                   <span class="st">&quot;COMP&quot;</span>, <span class="st">&quot;SFRP2&quot;</span>),
                 <span class="dt">cell_cycle=</span><span class="kw">c</span>(<span class="st">&quot;CCNA1&quot;</span>, <span class="st">&quot;CDKN1A&quot;</span>, <span class="st">&quot;CDKN2A&quot;</span>),
                 <span class="dt">inflam=</span><span class="kw">c</span>(<span class="st">&quot;CCL18&quot;</span>, <span class="st">&quot;CCL13&quot;</span>, <span class="st">&quot;C1QA&quot;</span>, <span class="st">&quot;C6&quot;</span>, <span class="st">&quot;C7&quot;</span>), <span class="co"># CCL18 is good for seperate control &amp; FSHD</span>
                 <span class="dt">immunoglobulin =</span> <span class="kw">c</span>(<span class="st">&quot;IGHG4&quot;</span>, <span class="st">&quot;IGHGP&quot;</span>, <span class="st">&quot;IGHG1&quot;</span>), 
                 <span class="dt">muscle =</span> <span class="kw">c</span>(<span class="st">&quot;PAX7&quot;</span>, <span class="st">&quot;PAX3&quot;</span>, <span class="st">&quot;MYOG&quot;</span>, <span class="st">&quot;MYOD1&quot;</span>)) </code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">marker_id &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(robust_markers, get_ensembl, sanitized.rlg))
data_lst &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">levels</span>(sanitized.rlg<span class="op">$</span>RNA_cluster_<span class="dv">5</span>), <span class="cf">function</span>(x, rlg) {
  s &lt;-<span class="st"> </span><span class="kw">colnames</span>(rlg)[rlg<span class="op">$</span>RNA_cluster_<span class="dv">5</span> <span class="op">==</span><span class="st"> </span>x]
  data &lt;-<span class="st"> </span><span class="kw">assay</span>(rlg[marker_id, s]) 
  <span class="co"># re-arrange by the colmeans (look heatmap)</span>
  Colv &lt;-<span class="st"> </span><span class="kw">colMeans</span>(data, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)
  data &lt;-<span class="st"> </span>data[, <span class="kw">order</span>(Colv)]
}, <span class="dt">rlg=</span>sanitized.rlg)
data &lt;-<span class="st"> </span><span class="kw">do.call</span>(cbind, data_lst)
<span class="kw">rownames</span>(data) &lt;-<span class="st"> </span><span class="kw">rowData</span>(sanitized.rlg[<span class="kw">rownames</span>(data)])<span class="op">$</span>gene_name
zscore_data &lt;-<span class="st"> </span>(data <span class="op">-</span><span class="st"> </span><span class="kw">rowMeans</span>(data)) <span class="op">/</span><span class="st"> </span><span class="kw">rowSds</span>(data)
gaps_row &lt;-<span class="st"> </span><span class="kw">cumsum</span>(<span class="kw">lapply</span>(robust_markers, length))

<span class="co">#&#39; annotation bar</span>
classes &lt;-<span class="st"> </span><span class="kw">as.character</span>(sanitized.rlg[, <span class="kw">colnames</span>(data)]<span class="op">$</span>RNA_cluster_<span class="dv">5</span>)
pheno_type &lt;-<span class="st"> </span>sanitized.dds[, <span class="kw">colnames</span>(data)]<span class="op">$</span>pheno_type                         
annotation_col &lt;-<span class="st"> </span>
<span class="st">  </span><span class="kw">data.frame</span>(<span class="dt">sample_name =</span> <span class="kw">colnames</span>(data),
             <span class="dt">classes     =</span> classes,
             <span class="dt">k_means     =</span> sanitized.dds[, <span class="kw">colnames</span>(data)]<span class="op">$</span>RNA_cluster_<span class="dv">5</span>,
             <span class="dt">FSHD =</span> sanitized.dds[, <span class="kw">colnames</span>(data)]<span class="op">$</span>pheno_type) 

new_time &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(mri_pathology<span class="op">$</span>time_<span class="dv">1</span>, mri_pathology<span class="op">$</span>time_<span class="dv">2</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(sample_name<span class="op">!=</span><span class="st">&quot;32-0008b&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample_name=</span><span class="kw">as.character</span>(sample_name))

new_time<span class="op">$</span>sample_name[new_time<span class="op">$</span>sample_name<span class="op">==</span><span class="st">&quot;32-0002b&quot;</span>] &lt;-<span class="st"> &quot;32-0002b1&quot;</span>
gaps_col &lt;-<span class="st"> </span><span class="kw">cumsum</span>(<span class="kw">count</span>(annotation_col, k_means)[, <span class="st">&quot;n&quot;</span>])<span class="op">$</span>n

annotation_col &lt;-<span class="st"> </span>annotation_col <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">left_join</span>(new_time, <span class="dt">by=</span><span class="st">&quot;sample_name&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">rename</span>(<span class="dt">Pathology=</span>Pathology.Score, <span class="dt">T1=</span>fat_fraction) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(sample_name, classes, FSHD, T1, STIR_rating, Inflammation, Pathology) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">column_to_rownames</span>(<span class="dt">var=</span><span class="st">&quot;sample_name&quot;</span>) </code></pre></div>
<pre><code>## Warning: Column `sample_name` joining factor and character vector, coercing
## into character vector</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">col &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;#f7f7f7&quot;</span>, <span class="st">&quot;#238b45&quot;</span>)
ann_cor &lt;-<span class="st"> </span><span class="kw">list</span>(
  <span class="dt">STIR_rating=</span> col,  <span class="dt">T1 =</span> col, <span class="dt">FSHD=</span><span class="kw">c</span>(<span class="dt">Control=</span><span class="st">&quot;white&quot;</span>, <span class="dt">FSHD=</span><span class="st">&quot;seagreen&quot;</span>),
  <span class="dt">Inflammation =</span> col, <span class="dt">Fibrosis =</span> col,
  <span class="dt">Var.in.Fiber.Size =</span> col, <span class="dt">Nec.Reg.Inflamm =</span> col,
  <span class="dt">Nucleation =</span> col, <span class="dt">Pathology =</span> col,
  <span class="dt">classes =</span> cluster_color)

<span class="co">#&#39; heatmap of robust markers</span>
<span class="kw">pheatmap</span>(zscore_data, <span class="dt">annotation_col=</span>annotation_col,
         <span class="dt">fontsize_row=</span><span class="dv">7</span>, <span class="dt">fontsize_col=</span><span class="dv">6</span>,
         <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, 
         <span class="dt">gaps_col=</span>gaps_col,
         <span class="dt">gaps_row=</span>gaps_row,
         <span class="dt">annotation_colors=</span>ann_cor,
         <span class="dt">cellheight=</span><span class="dv">10</span>, <span class="dt">annotation_legend=</span><span class="ot">FALSE</span>,
         <span class="dt">cluster_rows=</span><span class="ot">FALSE</span>, <span class="dt">cluster_cols=</span><span class="ot">FALSE</span>)</code></pre></div>
<p><img src="gitbook_files/figure-html/heatmap_mk_genes-1.png" width="672" /></p>
</div>
<div id="mri-characterisitcs-and-pathology-scores-by-fshd-classes" class="section level2">
<h2><span class="header-section-number">5.5</span> MRI characterisitcs and pathology scores by FSHD classes</h2>
<p>The chunk below summarizes the average MRI and pathology scores of each class.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">histo &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(mri_pathology<span class="op">$</span>time_<span class="dv">1</span>, mri_pathology<span class="op">$</span>time_<span class="dv">2</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample_name =</span> <span class="kw">as.character</span>(sample_name)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample_name =</span> <span class="kw">ifelse</span>(sample_name <span class="op">==</span><span class="st"> &quot;32-0002b&quot;</span>, <span class="st">&quot;32-0002b1&quot;</span>, sample_name)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">left_join</span>(cluster_df, <span class="dt">by=</span><span class="st">&quot;sample_name&quot;</span>)</code></pre></div>
<pre><code>## Warning: Column `sample_name` joining character vector and factor, coercing
## into character vector</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#&#39; sanity check: all cluster$sample_name are in time_var$sample_name</span>
<span class="kw">all</span>(cluster_df<span class="op">$</span>sample_name <span class="op">%in%</span><span class="st"> </span>histo<span class="op">$</span>sample_name)</code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#&#39; join cluster_df to histology/mri scores</span>
mri_stats &lt;-<span class="st"> </span>histo <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">group_by</span>(new_cluster_name)  <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">summarize</span>(<span class="dt">pathology_avg =</span> <span class="kw">mean</span>(Pathology.Score, <span class="dt">na.rm=</span><span class="ot">TRUE</span>),
            <span class="dt">pathology_sd =</span> <span class="kw">sd</span>(Pathology.Score, <span class="dt">na.rm=</span><span class="ot">TRUE</span>),
            <span class="dt">T1_fraction_avg =</span> <span class="kw">mean</span>(fat_fraction, <span class="dt">na.rm=</span><span class="ot">TRUE</span>),
            <span class="dt">T1_fraction_sd =</span> <span class="kw">sd</span>(fat_fraction, <span class="dt">na.rm=</span><span class="ot">TRUE</span>),
            <span class="dt">STIR_avg =</span> <span class="kw">mean</span>(STIR_rating),
            <span class="dt">STIR_sd =</span> <span class="kw">sd</span>(STIR_rating),
            <span class="dt">Inflam_avg =</span> <span class="kw">mean</span>(Inflammation, <span class="dt">na.rm=</span><span class="ot">TRUE</span>),
            <span class="dt">Infalm_se =</span> <span class="kw">mean</span>(Inflammation, <span class="dt">na.rm=</span><span class="ot">TRUE</span>))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(wesanderson)
col &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">wes_palette</span>(<span class="dt">n=</span><span class="dv">4</span>, <span class="dt">name=</span><span class="st">&quot;Cavalcanti1&quot;</span>, <span class="dt">type=</span><span class="st">&quot;discrete&quot;</span>))
tmp &lt;-<span class="st"> </span>mri_stats <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(new_cluster_name, pathology_avg, Inflam_avg, STIR_avg) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">rename</span>(<span class="dt">Pathology=</span>pathology_avg, <span class="dt">Inflamm=</span>Inflam_avg, <span class="dt">STIR=</span>STIR_avg) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">gather</span>(feature, average.score, <span class="op">-</span>new_cluster_name)
gg &lt;-<span class="st"> </span><span class="kw">ggplot</span>(tmp, <span class="kw">aes</span>(<span class="dt">x=</span>new_cluster_name, <span class="dt">y=</span>average.score, <span class="dt">fill=</span>feature)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">width=</span><span class="fl">0.7</span>)  <span class="op">+</span>
<span class="st">  </span><span class="co">#scale_fill_brewer(palette=&quot;Set2&quot;) +</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span>col) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme_bw</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">labs</span>(<span class="dt">y=</span><span class="st">&quot;Average score&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="fl">0.5</span>, <span class="dt">size=</span><span class="dv">9</span>),
        <span class="dt">legend.key.size =</span> <span class="kw">unit</span>(<span class="fl">0.8</span>, <span class="st">&quot;line&quot;</span>),
        <span class="co">#legend.position=&quot;bottom&quot;,</span>
        <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(),
        <span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">10</span>),
        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">25</span>, <span class="dt">hjust=</span><span class="dv">1</span>))
gg</code></pre></div>
<div class="figure">
<img src="gitbook_files/figure-html/viz_mri_stats-1.png" alt="Average MRI and pathology scores by FSHD classes." width="672" />
<p class="caption">
(#fig:viz_mri_stats)Average MRI and pathology scores by FSHD classes.
</p>
</div>
</div>
<div id="discriminative-biomarkers-characteristics-by-fshd-classes" class="section level2">
<h2><span class="header-section-number">5.6</span> Discriminative biomarkers characteristics by FSHD classes</h2>
<p>The code chunk below summarizes the average expression of five biological processes in FSHD relative to the controls.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(xlsx)
table_dir &lt;-<span class="st"> </span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;suppl_tables&quot;</span>)
file_name &lt;-<span class="st"> </span><span class="kw">file.path</span>(table_dir, <span class="st">&quot;suppl_table_5_Candidate_Biomarkers_and_Enriched_Go.xlsx&quot;</span>)
<span class="co">#&#39; High vs Control differential analysis sheet</span>
de &lt;-<span class="st"> </span><span class="kw">read.xlsx</span>(<span class="dt">file=</span>file_name, <span class="dt">sheetIndex=</span><span class="dv">1</span>, <span class="dt">startRow=</span><span class="dv">4</span>)
ecm_de &lt;-<span class="st"> </span>de <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(ecm <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pull</span>(gencode_id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.character</span>(.)
dux4_de &lt;-<span class="st"> </span>de <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(DUX4_induced <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pull</span>(gencode_id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.character</span>(.)
inflamm_de &lt;-<span class="st"> </span>de <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(inflamm <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span> <span class="op">|</span><span class="st"> </span>immune <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">pull</span>(gencode_id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.character</span>(.)
stress_de &lt;-<span class="st"> </span>de <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(stress <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pull</span>(gencode_id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.character</span>(.)
ig_de &lt;-<span class="st"> </span>de <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;IGH&quot;</span>, gene_name) <span class="op">|</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;IGK&quot;</span>, gene_name)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">pull</span>(gencode_id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.character</span>(.)
de_list &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">Extracellur_Matrix=</span>ecm_de, <span class="dt">DUX4=</span>dux4_de, <span class="dt">Inflammatory=</span>inflamm_de, <span class="dt">Immunoglobulin=</span>ig_de)

<span class="co">#&#39; add muscle</span>
muscle &lt;-<span class="st"> </span><span class="kw">c</span>(
  <span class="st">&quot;MYF5&quot;</span>, <span class="st">&quot;MYF6&quot;</span>, <span class="st">&quot;MYH1&quot;</span>, <span class="co"># newly added</span>
  <span class="st">&quot;MYH13&quot;</span>, <span class="st">&quot;MYH2&quot;</span>, <span class="st">&quot;MYH4&quot;</span>, <span class="st">&quot;MYH6&quot;</span>, <span class="st">&quot;MYH7&quot;</span>, <span class="co"># newly added</span>
  <span class="st">&quot;MYHAS&quot;</span>, <span class="st">&quot;MYL1&quot;</span>, <span class="st">&quot;MYL2&quot;</span>, <span class="co">#newly added</span>
  <span class="st">&quot;MYO18B&quot;</span>, <span class="st">&quot;MYOD1&quot;</span>, <span class="st">&quot;MYOG&quot;</span>, <span class="st">&quot;MYOM2&quot;</span>, <span class="st">&quot;MYOM3&quot;</span>,
  <span class="st">&quot;MYOT&quot;</span>, <span class="st">&quot;MYOZ1&quot;</span>, <span class="st">&quot;MYOZ2&quot;</span>, <span class="st">&quot;MYOZ3&quot;</span>, <span class="st">&quot;PAX7&quot;</span>, <span class="st">&quot;PAX3&quot;</span>)
de_list<span class="op">$</span>Muscle &lt;-<span class="st"> </span><span class="kw">unname</span>(<span class="kw">get_ensembl</span>(muscle, sanitized.rlg))

<span class="co"># average expression relative to controls</span>
de_avg_per_class &lt;-<span class="st"> </span><span class="kw">lapply</span>(de_list, <span class="cf">function</span>(id){
  tmp &lt;-<span class="st"> </span>sanitized.rlg[id, ]
  class &lt;-<span class="st"> </span><span class="kw">levels</span>(sanitized.rlg<span class="op">$</span>RNA_cluster_<span class="dv">5</span>)
  avg_rlg &lt;-<span class="st"> </span><span class="kw">sapply</span>(class, <span class="cf">function</span>(x){
    <span class="kw">colMeans</span>(<span class="kw">assay</span>(tmp)[, tmp<span class="op">$</span>RNA_cluster_<span class="dv">5</span> <span class="op">==</span><span class="st"> </span>x])
  })
  avg_per_class &lt;-<span class="st"> </span><span class="kw">sapply</span>(avg_rlg, mean)
  avg_per_class <span class="op">-</span><span class="st"> </span>avg_per_class[<span class="st">&quot;A_Cntr&quot;</span>]
})

de_avg_per_class &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">do.call</span>(cbind, de_avg_per_class)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var=</span><span class="st">&quot;class&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(class <span class="op">!=</span><span class="st"> &quot;A_Cntr&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">RNA_cluster =</span> <span class="kw">.face_off_cluster</span>(class)) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>class) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">gather</span>(feature, average.score, <span class="op">-</span>RNA_cluster) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">feature=</span><span class="kw">factor</span>(feature))

col &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">wes_palette</span>(<span class="dt">n=</span><span class="dv">5</span>, <span class="dt">name=</span><span class="st">&quot;Darjeeling1&quot;</span>, <span class="dt">type=</span><span class="st">&quot;discrete&quot;</span>))
gg &lt;-<span class="st"> </span><span class="kw">ggplot</span>(de_avg_per_class, <span class="kw">aes</span>(<span class="dt">x=</span>RNA_cluster, <span class="dt">y=</span>average.score, <span class="dt">fill=</span>feature)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">width=</span><span class="fl">0.75</span>)  <span class="op">+</span>
<span class="st">  </span><span class="co">#scale_fill_brewer(palette=&quot;Set2&quot;) +</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span>col) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme_bw</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">labs</span>(<span class="dt">y=</span><span class="st">&quot;Average expression - control&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="fl">0.5</span>, <span class="dt">size=</span><span class="dv">9</span>),
        <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(),
        <span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">9</span>),
        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">25</span>, <span class="dt">hjust=</span><span class="dv">1</span>, <span class="dt">size=</span><span class="dv">8</span>), 
        <span class="dt">legend.key.size =</span> <span class="kw">unit</span>(<span class="fl">0.75</span>, <span class="st">&quot;line&quot;</span>))
gg</code></pre></div>
<p><img src="gitbook_files/figure-html/class_avg_expression-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="longitudinal-changes.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="mild-fshd.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["gitbook.pdf", "gitbook.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
